# (PART\*) Read approach {-}

# Taxonomic profiling {#taxprofile}
```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/classification.png", auto_pdf = TRUE)
``` 

There are a number of methods for determining the species composition of a metagenomic data-set, but for the purposes of this practical we will use `Kraken2` & `Bracken` (Bayesian Reestimation of Abundance with KrakEN).  `Kraken2` classifies short DNA with taxonomic labels and is frequently used for metagenomic studies. `Bracken` uses the taxonomic labels assigned by `Kraken2` to compute the abundance of species in a set of DNA sequences.

First, we'll make a new directory for it and move into it, after returning home:

```{bash eval=FALSE}
cd ..
mkdir 3-Taxonomy
cd 3-Taxonomy
```

## Kraken2
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/kraken.png", auto_pdf = TRUE)
``` 

Prior to running `Kraken2` we need to set a variable so `Kraken2` knows where to look for the databases it will use.

```{bash eval=FALSE}
export KRAKEN2_DB_PATH=/pub39/tea/matthew/NEOF/Shotgun_metagenomics/kraken2_db
```

__Note__: You can look at the contents of the above directory to see it currently contains the MiniKraken database. This database contains only a subset of the bacteria, archaea, and viral `Kraken2` libraries. This is used in this practical due to restrictions on time and computational resources. For your own analyses we would recommend the full `Kraken2` database which uses all the bacteria, achaeal and viral complete genomes that are in Refseq at the time of building. See the appendix for a link to the `Kraken2` manual.

### Kraken2: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/enter_key.png", auto_pdf = TRUE)
``` 

Now, run `Kraken2` on sample K1 by running the following command.

__Note__: As there is little human sequence in our data we will not use host removed data.

```{bash eval=FALSE}
kraken2 --paired --db minikraken2_v1_8GB \
--output K1.kraken --report K1.kreport2 \
~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz
```  

While this is running, let's look at what those command line options do:

- `~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz` : the trimmed read pairs for K1, which we will use as input.
- `--paired` : Indicates that we are providing paired reads to `Kraken2`. Internally, `Kraken2` will concatenate the R1 and R2 reads into one sequence with an N between them.
- `--db` : Specify the `Kraken2` database to be used for taxonomic classification. Previous to the command we set the `KRAKEN_DB_PATH` so in this case the command will look for the directory called `minikraken2_v1_8GB` within `KRAKEN_DB_PATH`. Alternatively the full path of the required database could be provided.
- `--threads` : How many CPUs the process will use.
- `--output` : This is the output file. 

### Kraken2: output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/output.png", auto_pdf = TRUE)
``` 

There are two major output formats from Kraken2:

- `--output`, `.kraken`: Each sequence (or sequence pair, in the case of paired reads) classified by `Kraken2` results in a single line of output. `Kraken2`'s output lines contain five tab-delimited fields; from left to right, they are:
   1. "C"/"U": a one letter code indicating that the sequence was either classified or unclassified.
   2. The sequence ID, obtained from the FASTA/FASTQ header.
   3. The taxonomy ID `Kraken2` used to label the sequence; this is 0 if the sequence is unclassified.
   4. The length of the sequence in bp. In the case of paired read data, this will be a string containing the lengths of the two sequences in bp, separated by a pipe character, e.g. "98|94".
   5. A space-delimited list indicating the LCA mapping of each k-mer in the sequence(s). For example, "562:13 561:4 A:31 0:1 562:3" would indicate that:
      - the first 13 k-mers mapped to taxonomy ID #562
      - the next 4 k-mers mapped to taxonomy ID #561
      - the next 31 k-mers contained an ambiguous nucleotide
      - the next k-mer was not in the database
      - the last 3 k-mers mapped to taxonomy ID #562
      - __Note__: that paired read data will contain a "|:|" token in this list to indicate the end of one read and the beginning of another.

- `--report`, `.kreport2`: The report output format. This is required for bracken. It is tab-delimited with one line per taxon. The fields of the output, from left-to-right, are as follows:
   1. Percentage of fragments covered by the clade rooted at this taxon.
   2. Number of fragments covered by the clade rooted at this taxon.
   3. Number of fragments assigned directly to this taxon.
   4. A rank code, indicating (U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. Taxa that are not at any of these 10 ranks have a rank code that is formed by using the rank code of the closest ancestor rank with a number indicating the distance from that rank. E.g., "G2" is a rank code indicating a taxon is between genus and species and the grandparent taxon is at the genus rank.
   5. NCBI taxonomic ID number
   6. Indented scientific name

The output to screen will show how many sequences are classified. This will be lower than normal as we are using a mini `Kraken2` database.

In a real analysis you may use the option `--confidence` which represents  the __"Confidence score threshold"__. The default is `0.0`, which is the lowest, with the maximum value being `1`. A good place to start may be `0.1`. Too many classifications are removed if you attempt it with this dataset, due to the mini `Kraken2` database used. More info on the confidence scoring can be found at: https://github.com/DerrickWood/kraken2/wiki/Manual#confidence-scoring

__Task__: Once the Kraken2 command has finished running, run it on the other two samples. 

__Hint__: You will need to change all instances of K1 to K2 or W1 in the above command

## Krona plot
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/krona.png", auto_pdf = TRUE)
``` 

`Krona` is an interactive metagenome species abundance visualisation tool. 

We can use the `Kraken2` report files to create our Krona plots. With the below command we can import our `Kraken2` taxonomy (within the report file) into a `Krona` html.

```{bash eval=FALSE}
ktImportTaxonomy -o kraken2.krona.html *.kreport2
```

`-o` is our output html file, and the final argument `*.kreport2` represents all of our `.kreport2` files in the current directory. The `*` is a wild-card, meaning any characters any number of times. Therefore `*.kreport2` identifies the files `K1.kreport2` `K2.kreport2` and `W1.kreport2`.

You will get a warning that not all taxonomy IDs were found. We will ignore this but in your own future installations this should be addressed with `Krona`'s `updateTaxonomy.sh` command.

Now we can view our interactive chart in a web browser.

```{bash eval=FALSE}
firefox kraken2.krona.html &
```

__Question__: Can you tell which sample looks the most different in terms of bacterial species that are present and absent?

## Bracken
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/bracken.png", auto_pdf = TRUE)
```

`Bracken` (Bayesian Reestimation of Abundance with KrakEN) uses taxonomy labels assigned by `Kraken2` to compute estimated abundances of species in a metagenomic sample. 

### Bracken: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/enter_key.png", auto_pdf = TRUE)
``` 

Just like with `Krona` we can use the `Kraken2` report files to run bracken.

```{bash eval=FALSE}
bracken -d $KRAKEN2_DB_PATH/minikraken2_v1_8GB \
-i K1.kreport2 -o K1.bracken -r 100 -l S -t 5
```

Let's look at what those command line options do:

- `-d` : Specifies the `Kraken2` database that was used for taxonomic classification. In this case bracken requires the variable `$KRAKEN_DB_PATH` so the option is provided the full path to the kraken database. For clarity try the command `ls $KRAKEN2_DB_PATH/minikraken2_v1_8GB`. 
- `-i` : The `Kraken2` report file, this will be used as the input.
- `-o` : The output `Bracken` file. Information about its contents is below.
- `-r 100`: This is the ideal length of the reads that were used in the `Kraken2` classification. It is recommended that the initial read length of the sequencing data is used. We are using 100 here as we used a paired library of 100 bp reads.
- `-l S`: This specifies the taxonomic level/rank of the `Bracken` output. In this case `S` is equal to species with the other options being `D`, `P`, `C`, `O`,`F` and `G`.
- `-t 5`: This specifies the minimum number of reads required for a classification at the specified rank. Any classifications with fewer reads than the specified threshold will not receive additional reads from higher taxonomy levels when distributing reads for abundance estimation. Five has been chosen here for this example data but in real datasets you may want to increase this number (default is 10).

### Bracken: output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/output.png", auto_pdf = TRUE)
``` 

The output file of `Bracken` contains the following columns:

1. __Name__: Name of taxonomy at the specified taxonomic level.
2. __Taxonomy ID__: NCBI taxonomy id
3. __Level ID__: Letter signifying the taxonomic level of the classification
4. __Kraken assigned read__: Number of reads assigned to the taxonomy by `Kraken2`.
5. __Added reads with abundance reestimation__: Number of reads added to the taxonomy by Bracken abundance reestimation. 
6. __Total reads after abundance reestimation__: Number from field 4 and 5 summed. This is the field that will be used for downstream analysis.
7. __Fraction of total reads__: Relative abundance of the taxonomy.

Use `less` or `vim` to look at the bracken output.

__Task__: Repeat the above commands for K2 and W1

### Bracken: merging output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/merge.png", auto_pdf = TRUE)
``` 

To make full use of `Bracken` output, it is best to merge the output into one table. However before we do this we’ll copy the `Bracken` output of other samples that have been generated prior to the workshop. These are all either Korean or Western Diet samples.

```{bash eval=FALSE}
cp /pub39/tea/matthew/NEOF/Shotgun_metagenomics/bracken/* .
```

Now to merge all the K and W `Bracken` files.

```{bash eval=FALSE}
combine_bracken_outputs.py --files [KW]*.bracken -o all.bracken
```

This output file contains the first three columns:

- __name__ = Organism group name. This will be based on the TAX_LVL chosen in the `Bracken` command and will only show the one level.
- __taxonomy_id__ = Taxonomy id number.
- __taxonomy_lvl__ = A single string indicating the taxonomy level of the group. ('D','P','C','O','F','G','S').

Following these columns are the following two columns for each sample.

- `${SampleName}.bracken_num`: The number of reads after abundance reestimation 
- `${SampleName}.bracken_frac`: Relative abundance of the group in the sample

### Bracken: extracting output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/exctractor.png", auto_pdf = TRUE)
``` 

We want a file with only the first column (organism name) and the `bracken_num` columns for each sample. To carry this out we  first create a sequence of numbers that will match the `bracken_num` column numbers. These start at column 4 and are every even numbered column after this. We will use `seq` to create every `2` numbers from the numbers `4` to `50` with commas (`,`) as separators (`-s`). 

__Note__: The number 50 is chosen as 3 (first three info columns) + 24*2 (24 samples with 2 columns each) = 50.

```{bash eval=FALSE}
#Try out the seq command to see its output
seq -s , 4 2 50
#Create variable
bracken_num_columns=$(seq -s , 4 2 50)
echo $bracken_num_columns
```

Now to use the variable to extract the `bracken_num` columns plus the first column (species names).

```{bash eval=FALSE}
cat all.bracken | cut -f 1,$bracken_num_columns > all_num.bracken
```

## LEfSe biomarker detection
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/lefse.png", auto_pdf = TRUE)
```

We will use `LEfSe` (Linear discriminant analysis Effect Size) to determine which taxa can most likely explain the differences between the Western and Korean diet. `LEfSe` couples standard tests for statistical significance with additional tests encoding biological consistency and effect relevance. It can be used with other features such as organisms, clades, operational taxonomic units, genes, or functions.

In essence it allows for the detection of biomarkers when comparing sample groups. In the `LEfSe` terminology the sample groups are called the class.

### LEfSe: add metadata
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/header.png", auto_pdf = TRUE)
```

We need to add metadata to our `Bracken` file to be ready for `LEfSe.` First we will copy the file so we have a backup in case we do anything wrong.

```{bash eval=FALSE}
cp all_num.bracken all_num.lefse.bracken
```

Using your favourite text editor (e.g. `nano`, `vim`, etc.) add the following line to the top of your `all_num.lefse.bracken` file. The words are separated by tabs. If you are not sure how to carry out this task please ask a demonstrator.

__diet	K	K	K	K	K	K	K	K	K	K	K	K	W	W	W	W	W	W	W	W	W	W	W	W__

__Note__: The above is __diet__ followed by 12 __K__ and 12 __W__.

The singular line should match the order of your samples within the file. This is the metadata line that `LEfSe` will use to determine which samples belong to each sample group, and therefore which to compare. In this case it is Korean diet samples versus Western diet samples.

### LEfSe: conda
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
```

`LEfSe` requires python2 whilst we have been using packages that required python3. We therefore need to use a different `conda` environment. We are currently using the `conda` env called `shotgun_meta` as represented by `(shotgun_meta)`. This was activated with the command `. useshotgun`.

Open a new terminal (right click on the main screen background, choose `Applications` -> `Shell` -> `bash`) and run the following in the new terminal to activate the `lefse` `conda` environment.

```{bash eval=FALSE}
#Setup environment
. uselefse
#Change directory
cd ~/3-Taxonomy
```

### LEfSe: format
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/lefse_format.png", auto_pdf = TRUE)
```

We need to further format and preprocess our file with a `LEfSe` script.

```{bash eval=FALSE}
lefse-format_input.py all_num.lefse.bracken all_num.lefse -c 1 -u 2 -o 1000000
```

- `all_num.lefse.bracken` : Input `Bracken` file.
- `all_num.lefse` : Output file formatted for the run_lefse command, which we will soon run.
- `-c 1` : Specifies the row with the class info. This is used to determine which samples will be compared against which samples. In this case it is the first row with the Ks and Ws.
- `-u 2` : Specifies the row with the sample names. This is the second row in this case.
- `-o 1000000` : An integer can be indicated to determine to what size (count sum value) each sample should be normalised to. `LEfSe` developers recommend 1000000 (1 million) when very low values a present. We generally always use 1 million for consistency.

### LEfSe: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/enter_key.png", auto_pdf = TRUE)
``` 

Now to run `LEfSe.` All we need to do is run the command with the formatted input and provide an output file name.

```{bash eval=FALSE}
run_lefse.py all_num.lefse all_num.lefse.out
```

The output file is a tab-delimited file which contains a row for each species. Biomarkers will have the five columns below whilst non-biomarkers will have the first two followed by a "-" .

- __Biomarker name__
- __Log of highest class average__: I.e. get the class with the greater amounts of the biomarker, average the counts and then get the log of this value.
- __Class with the greater amounts of biomarker__
- __LDA effect size__: A statistical figure for `LEfSe`..
- p-value: Biomarkers must have a p-value of <0.05 to be considered significant.

The __LDA effect__ size indicates how much of an effect each biomarker has. The default is to only count a species with an LDA effect size of greater than 2 or less than -2 as a biomarker. The further the LDA effect size is from 0 the greater the effect the species causes.

### LEfSe: visualisation
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/bar_chart_horizontal.png", auto_pdf = TRUE)
``` 

Next we can visualise the output.

```{bash eval=FALSE}
lefse-plot_res.py --dpi 200 --format png all_num.lefse.out biomarkers.png
```

- `--dpi 200` : Dots per inch. This refers to the resolution of the output image. Normally publications want 300 dpi. We’ve chosen 200 as it is good quality and we will not be publishing these results.
- `--format png` : Format of output file. png is a commonly used file format for images.
- `all_num.lefse.out` : `LEfSe` output to visualise.
- `biomarkers.png` : Plot showing the LDA scores of the species detected as biomarkers. Colouring shows which class (K or W) the species is found in higher abundance.

Look at the figure with the program `okular`:

```{bash eval=FALSE}
okular biomarkers.png
```

__Questions__:

- Which species causes the biggest effect in the W class and in the K class?
- Which class has more biomarkers associated with it?

__Note__: In this instance green bars represent biomarkers in higher abundance in the W samples whilst the red bars represent biomarkers in higher abundance in the K samples.

## Kraken2 and Bracken databases
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/database.png", auto_pdf = TRUE)
``` 

In your own future analysis you will need to create your own `Kraken2` and `Bracken` databases. Please sse the following links on information for this:

- `Kraken2`
   - Standard `Kraken2` databases: https://github.com/DerrickWood/kraken2/wiki/Manual#standard-kraken-2-database
   - Custom `Kraken2` databases: https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases
- `Bracken`
   - https://ccb.jhu.edu/software/bracken/index.shtml?t=manual#step1
   - This requires a `Kraken2` database to be built first.

