# Kraken2
```{r, fig.align = 'center',out.width= '150px', echo=FALSE }
knitr::include_graphics(path = "figures/kraken.png", auto_pdf = TRUE)
``` 

Prior to running `Kraken2` we need to set a variable so `Kraken2` knows where to look for the databases it will use.

```{bash eval=FALSE}
export KRAKEN2_DB_PATH=/pub14/tea/nsc206/NEOF/Shotgun_metagenomics/kraken2_db
```

__Note__: You can look at the contents of the above directory to see it currently contains the MiniKraken database. This database contains only a subset of the bacteria, archaea, and viral `Kraken2` libraries. This is used in this practical due to restrictions on time and computational resources. For your own analyses we would recommend the full `Kraken2` database which uses all the bacteria, achaeal and viral complete genomes that are in Refseq at the time of building. See the following links for info on installing the databases.

- Standard `Kraken2` databases: https://github.com/DerrickWood/kraken2/wiki/Manual#standard-kraken-2-database
- Custom `Kraken2` databases: https://github.com/DerrickWood/kraken2/wiki/Manual#custom-databases

## Kraken2: run
```{r, fig.align = 'center',out.width= '100px', echo=FALSE }
knitr::include_graphics(path = "figures/play_blue.png", auto_pdf = TRUE)
``` 

Now, run `Kraken2` on sample K1 by running the following command.

__Note__: We are not using the host removed data. This is to save time. In your own analysis ensure you are using host removed data.

```{bash eval=FALSE}
kraken2 --paired --db minikraken2_v1_8GB \
--output K1.kraken --report K1.kreport2 \
~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz
```  

#### Parameters {-}
```{r, fig.align = 'center',out.width= '100px', echo=FALSE }
knitr::include_graphics(path = "figures/parameter_blue.png", auto_pdf = TRUE)
```

- __`--paired`__: Indicates that we are providing paired reads to `Kraken2`. Internally, `Kraken2` will concatenate the R1 and R2 reads into one sequence with an N between them.
- __`--db`__: Specify the `Kraken2` database to be used for taxonomic classification. 
  - Previous to the command we set the `KRAKEN_DB_PATH` so in this case the command will look for the directory called `minikraken2_v1_8GB` within `KRAKEN_DB_PATH`. 
  - Alternatively the full path of the required database could be provided (`/pub39/tea/nsc006/NEOF/Shotgun_metagenomics/kraken2_db/minikraken2_v1_8GB`).
- __`--threads`__: Number of CPUs the process will use.
- __`--output`__: The output file. More info below.
- __`--report`__: The output report file. More info below.
- __`~/2-Trimmed/K1_R1.fq.gz ~/2-Trimmed/K1_R2.fq.gz`__: The trimmed read pairs for K1, which we will use as input.

## Kraken2: output
```{r, fig.align = 'center',out.width= '100px', echo=FALSE }
knitr::include_graphics(path = "figures/output_file_blue.png", auto_pdf = TRUE)
``` 

There are two major output formats from Kraken2:

- `--output`, `.kraken`: Each sequence (or sequence pair, in the case of paired reads) classified by `Kraken2` results in a single line of output. `Kraken2`'s output lines contain five tab-delimited fields; from left to right, they are:
   1. "C"/"U": a one letter code indicating that the sequence was either classified or unclassified.
   2. The sequence ID, obtained from the FASTA/FASTQ header.
   3. The taxonomy ID `Kraken2` used to label the sequence; this is 0 if the sequence is unclassified.
   4. The length of the sequence in bp. In the case of paired read data, this will be a string containing the lengths of the two sequences in bp, separated by a pipe character, e.g. "98|94".
   5. A space-delimited list indicating the LCA mapping of each k-mer in the sequence(s). For example, "562:13 561:4 A:31 0:1 562:3" would indicate that:
      - the first 13 k-mers mapped to taxonomy ID #562
      - the next 4 k-mers mapped to taxonomy ID #561
      - the next 31 k-mers contained an ambiguous nucleotide
      - the next k-mer was not in the database
      - the last 3 k-mers mapped to taxonomy ID #562
      - __Note__: that paired read data will contain a "|:|" token in this list to indicate the end of one read and the beginning of another.

- `--report`, `.kreport2`: The report output format. This is required for bracken. It is tab-delimited with one line per taxon. The fields of the output, from left-to-right, are as follows:
   1. Percentage of paired reads covered by the clade rooted at this taxon.
   2. Number of paired reads covered by the clade rooted at this taxon.
   3. Number of paired reads assigned directly to this taxon.
   4. A rank code, indicating (U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. Taxa that are not at any of these 10 ranks have a rank code that is formed by using the rank code of the closest ancestor rank with a number indicating the distance from that rank. E.g., "G2" is a rank code indicating a taxon is between genus and species and the grandparent taxon is at the genus rank.
   5. NCBI taxonomic ID number
   6. Indented scientific name

The output to screen will show how many sequences are classified. This will be lower than normal as we are using a mini `Kraken2` database.

In a real analysis you may use the option `--confidence` which represents  the __"Confidence score threshold"__. The default is `0.0`, which is the lowest, with the maximum value being `1`. A good place to start may be `0.1`. Too many classifications are removed if you attempt it with this dataset, due to the mini `Kraken2` database used. More info on the confidence scoring can be found at: https://github.com/DerrickWood/kraken2/wiki/Manual#confidence-scoring

__Task__: Once the Kraken2 command has finished running, run it on the other two samples. Attempt the commands without looking at the help box.

__Hint__: You will need to change all instances of K1 to K2 or W1 in the above command

`r hide("K2 & W1 Kraken 2 commands")`
```{bash eval=FALSE}
#K2
kraken2 --paired --db minikraken2_v1_8GB \
--output K2.kraken --report K2.kreport2 \
~/2-Trimmed/K2_R1.fq.gz ~/2-Trimmed/K2_R2.fq.gz
#W1
kraken2 --paired --db minikraken2_v1_8GB \
--output W1.kraken --report W1.kreport2 \
~/2-Trimmed/W1_R1.fq.gz ~/2-Trimmed/W1_R2.fq.gz
```
`r unhide()`

## Kraken 2: MCQs
```{r, fig.align = 'center',out.width= '100px', echo=FALSE }
knitr::include_graphics(path = "figures/question_bubble_blue.png", auto_pdf = TRUE)
```

Viewing the Kraken2 output files with your favourite text viewer (`less`, `nano`, `vim`, etc.), attempt the below MCQs.

```{r, echo = FALSE}
opts_p <- c("__2__", "__486,723__", answer="__555,162__")
```
1. How many paired reads were unclassified for K1? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__2__", answer="__486,723__", "__555,162__")
```
2. How many paired reads were classified for K2 (i.e. number of reads classified at root level and below)? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__2__", "__486,723__", "__555,162__")
```
3. How many paired reads were assigned directly to root level for W1? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__0.12__", "__0.59__", answer="__14.89__")
```
4. What percentage of W1's paired reads were assigned to the clade of Bacteroidetes (Phylum)? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__0.12__", answer="__0.59__", "__14.89__")
```
5. What percentage of K2's paired reads were assigned to the clade of Rikenellaceae (Family)? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__0.12__", "__0.59__", "__14.89__")
```
6. What percentage of K1's paired reads were assigned to the clade of Bacteroides helcogenes (Species)? `r longmcq(opts_p)`