# Krona plot
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/krona.png", auto_pdf = TRUE)
``` 

`Krona` is an interactive metagenome species abundance visualisation tool. 

## Krona: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/play_red.png", auto_pdf = TRUE)
``` 

We can use the `Kraken2` report files to create our Krona plots. With the below command we can import our `Kraken2` taxonomy (within the report file) into a `Krona` html.

```{bash eval=FALSE}
ktImportTaxonomy -o kraken2.krona.html *.kreport2
```

`-o` is our output html file, and the final argument `*.kreport2` represents all of our `.kreport2` files in the current directory. The `*` is a wild-card, meaning any characters any number of times. Therefore `*.kreport2` identifies the files `K1.kreport2` `K2.kreport2` and `W1.kreport2`.

You will get a warning that not all taxonomy IDs were found. We will ignore this but in your own future installations this should be addressed with `Krona`'s `updateTaxonomy.sh` command.

## Krona: visualise
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/eye_red.png", auto_pdf = TRUE)
``` 

We can view our interactive chart in a web browser. This shows the percentage of reads that were classified to various taxonomies at different levels.

```{bash eval=FALSE}
firefox kraken2.krona.html &
```

This is an interactive multi-tiered pie chart with many options. Some of the interactivity is described below:

- To choose a sample to view, click on the sample name in the top left.
- To zoom into a lower taxonomy, double click on the taxonomy's name on the pie chart.
  - E.g. double click the word "Bacteria" on the pie chart to zoom into Bacteria and therefore ignore Eukaryota, and "[other Root]".
- To zoom out to a higher taxonomy, click on the taxonomy's name in the middle of the pie chart.
  - E.g. click the word "root" in the middle of the pie chart to zoom back out from any lower level of taxonomy.
- To view percentage stats of a taxonomy, click on the taxonomy name on the pie chart. 
  - This will display the percentage this taxonomy covers of all the different taxonomies above it that it belongs to. This is displayed on the right side of the screen.
  - E.g. Click on Proteobacteria to see the percentage of the Root it accounts for and to see the percentage of Bacteria it accounts for in the sample.

## Krona: MCQs
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/question_bubble_red.png", auto_pdf = TRUE)
```

Viewing the Krona, attempt the below MCQs.

```{r, echo = FALSE}
opts_p <- c("__43%__", answer="__35%__", "__14%__")
```
1. What percentage of the __root__ was classified to __Bacteria__ for __K1__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__43%__", "__35%__", "__14%__")
```
2. What percentage of __bacteria__ was classified to __Proteobacteria__ for __K2__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__43%__", "__35%__", answer="__14%__")
```
3. What percentage of the __root__ was classified to __Azorhizobium__ for __W1__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__39%__", "__27%__", "__6%__")
```
4. What percentage of the __Terrabacteria groups__ was classified to __Cellulomonas gilvus__ for __K2__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__39%__", "__27%__", answer="__6%__")
```
5. What percentage of __Alphaproteobacteria__ was classified to __Phenylobacterium immobile__ for __K1__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__39%__", answer="__27%__", "__6%__")
```
6. What percentage of __Viridiplantae__ (in Eukaryota) was classified to __Zygnemataceae__ for __W1__?  `r longmcq(opts_p)`