# Bracken
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/bracken.png", auto_pdf = TRUE)
```

`Bracken` (Bayesian Reestimation of Abundance with KrakEN) uses taxonomy labels assigned by `Kraken2` to compute estimated abundances of species in a metagenomic sample. 

### Bracken: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/enter_key.png", auto_pdf = TRUE)
``` 

Just like with `Krona` we can use the `Kraken2` report files to run bracken.

```{bash eval=FALSE}
bracken -d $KRAKEN2_DB_PATH/minikraken2_v1_8GB \
-i K1.kreport2 -o K1.bracken -r 100 -l S -t 5
```

Let's look at what those command line options do:

- `-d` : Specifies the `Kraken2` database that was used for taxonomic classification. In this case bracken requires the variable `$KRAKEN_DB_PATH` so the option is provided the full path to the kraken database. For clarity try the command `ls $KRAKEN2_DB_PATH/minikraken2_v1_8GB`. 
- `-i` : The `Kraken2` report file, this will be used as the input.
- `-o` : The output `Bracken` file. Information about its contents is below.
- `-r 100`: This is the ideal length of the reads that were used in the `Kraken2` classification. It is recommended that the initial read length of the sequencing data is used. We are using 100 here as we used a paired library of 100 bp reads.
- `-l S`: This specifies the taxonomic level/rank of the `Bracken` output. In this case `S` is equal to species with the other options being `D`, `P`, `C`, `O`,`F` and `G`.
- `-t 5`: This specifies the minimum number of reads required for a classification at the specified rank. Any classifications with fewer reads than the specified threshold will not receive additional reads from higher taxonomy levels when distributing reads for abundance estimation. Five has been chosen here for this example data but in real datasets you may want to increase this number (default is 10).

## Bracken: output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/output.png", auto_pdf = TRUE)
``` 

The output file of `Bracken` contains the following columns:

1. __Name__: Name of taxonomy at the specified taxonomic level.
2. __Taxonomy ID__: NCBI taxonomy id
3. __Level ID__: Letter signifying the taxonomic level of the classification
4. __Kraken assigned read__: Number of reads assigned to the taxonomy by `Kraken2`.
5. __Added reads with abundance reestimation__: Number of reads added to the taxonomy by Bracken abundance reestimation. 
6. __Total reads after abundance reestimation__: Number from field 4 and 5 summed. This is the field that will be used for downstream analysis.
7. __Fraction of total reads__: Relative abundance of the taxonomy.

Use `less` or `vim` to look at the bracken output.

__Task__: Repeat the above commands for K2 and W1

## Bracken: merging output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/merge.png", auto_pdf = TRUE)
``` 

To make full use of `Bracken` output, it is best to merge the output into one table. However before we do this weâ€™ll copy the `Bracken` output of other samples that have been generated prior to the workshop. These are all either Korean or Western Diet samples.

```{bash eval=FALSE}
cp /pub39/tea/nsc006/NEOF/Shotgun_metagenomics/bracken/* .
```

Now to merge all the K and W `Bracken` files.

```{bash eval=FALSE}
combine_bracken_outputs.py --files [KW]*.bracken -o all.bracken
```

This output file contains the first three columns:

- __name__ = Organism group name. This will be based on the TAX_LVL chosen in the `Bracken` command and will only show the one level.
- __taxonomy_id__ = Taxonomy id number.
- __taxonomy_lvl__ = A single string indicating the taxonomy level of the group. ('D','P','C','O','F','G','S').

Following these columns are the following two columns for each sample.

- `${SampleName}.bracken_num`: The number of reads after abundance reestimation 
- `${SampleName}.bracken_frac`: Relative abundance of the group in the sample

## Bracken: extracting output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/exctractor.png", auto_pdf = TRUE)
``` 

We want a file with only the first column (organism name) and the `bracken_num` columns for each sample. To carry this out we  first create a sequence of numbers that will match the `bracken_num` column numbers. These start at column 4 and are every even numbered column after this. We will use `seq` to create every `2` numbers from the numbers `4` to `50` with commas (`,`) as separators (`-s`). 

__Note__: The number 50 is chosen as 3 (first three info columns) + 24*2 (24 samples with 2 columns each) = 50.

```{bash eval=FALSE}
#Try out the seq command to see its output
seq -s , 4 2 50
#Create variable
bracken_num_columns=$(seq -s , 4 2 50)
echo $bracken_num_columns
```

Now to use the variable to extract the `bracken_num` columns plus the first column (species names).

```{bash eval=FALSE}
cat all.bracken | cut -f 1,$bracken_num_columns > all_num.bracken
```