# Appendix
```{r, fig.align = 'center',out.width= '25%', echo=FALSE }
knitr::include_graphics(path = "figures/appendix.png", auto_pdf = TRUE)
``` 

## Next steps
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/step.png", auto_pdf = TRUE)
``` 

Below are some good links to start with before carrying out your own projects.

- A good review paper on metagenome assembly approaches
   - Genome-resolved metagenomics using environmental and clinical samples: https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbab030/6184411
- bioBakery tools for meta'omic profiling
   - https://github.com/biobakery/biobakery
- Assessing the performance of different approaches for functional and taxonomic annotation of metagenomes
   - https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6289-6
- MetaEuk: Functional annotation of eukaryotic metagenome
   - https://github.com/soedinglab/metaeuk
_ Methods for Metagenomic data visualisation and analysis
   - https://www.researchgate.net/publication/318252633_Methods_for_The_Metagenomic_Data_Visualization_and_Analysis




## Manuals
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/manual.png", auto_pdf = TRUE)
``` 

Conda: https://conda.io/projects/conda/en/latest/user-guide/getting-started.html

FastQC: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

MultiQC: https://multiqc.info/

Trim Galore: https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/

Bowtie2: http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml

samtools: http://www.htslib.org/

BBTools: https://jgi.doe.gov/data-and-tools/bbtools/

Kraken2: https://github.com/DerrickWood/kraken2/wiki/Manual

Krona: https://github.com/marbl/Krona/wiki/KronaTools

Bracken: https://ccb.jhu.edu/software/bracken/index.shtml?t=manual

LEfSe: https://huttenhower.sph.harvard.edu/lefse/

HUMAnN2: https://github.com/biobakery/biobakery/wiki/humann2

MetaPhlAn2: https://huttenhower.sph.harvard.edu/metaphlan2/

Biobakery: https://github.com/biobakery/biobakery

MegaHit: https://github.com/voutcn/megahit

BWA: https://github.com/lh3/bwa

minimap2: https://github.com/lh3/minimap2

MetaBAT2: https://bitbucket.org/berkeleylab/metabat/src/master/

CheckM: https://github.com/Ecogenomics/CheckM/wiki

PhyloPhlAn: https://github.com/biobakery/phylophlan/wiki

MetaWRAP: https://github.com/bxlab/metaWRAP

## COG Cateories

- CELLULAR PROCESSES AND SIGNALING
   - [D] Cell cycle control, cell division, chromosome partitioning
   - [M] Cell wall/membrane/envelope biogenesis
   - [N] Cell motility
   - [O] Post-translational modification, protein turnover, and chaperones
   - [T] Signal transduction mechanisms
   - [U] Intracellular trafficking, secretion, and vesicular transport
   - [V] Defense mechanisms
   - [W] Extracellular structures
   - [Y] Nuclear structure
   - [Z] Cytoskeleton
- INFORMATION STORAGE AND PROCESSING
   - [A] RNA processing and modification
   - [B] Chromatin structure and dynamics
   - [J] Translation, ribosomal structure and biogenesis
   - [K] Transcription
   - [L] Replication, recombination and repair
- METABOLISM
   - [C] Energy production and conversion
   - [E] Amino acid transport and metabolism
   - [F] Nucleotide transport and metabolism
   - [G] Carbohydrate transport and metabolism
   - [H] Coenzyme transport and metabolism
   - [I] Lipid transport and metabolism
   - [P] Inorganic ion transport and metabolism
   - [Q] Secondary metabolites biosynthesis, transport, and catabolism
- POORLY CHARACTERIZED
   - [R] General function prediction only
   - [S] Function unknown
   
## Obtaining Read Data

The following commands can be used to obtain the sequence data used in this practical, directly from the EBI metagenomics site. It is worth noting that these are the full set of data, not like the miniaturised version you have used in the tutorial.

```{bash eval=FALSE}
wget -O K1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_1.fastq.gz
wget -O K1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_2.fastq.gz
wget -O K2_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_1.fastq.gz
wget -O K2_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_2.fastq.gz
wget -O W1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_1.fastq.gz
wget -O W1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_2.fastq.gz
```
