# Assembly functional annotation {#funcanno}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/pathways.png", auto_pdf = TRUE)
``` 

## Taxonomic annotation
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/classification.png", auto_pdf = TRUE)
``` 

Taxonomic annotation of bins can be carried out with `Kraken2`. As we have already done this for the reads and taxonomic results from read and assembly approaches have similar performances we will not cover it here, and instead move straight onto functional annotation with `Prokka`.

## Prokka
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/quokka.png", auto_pdf = TRUE)
``` 

We will carry out `Prokka` functional annotation using `MetaWRAP`. `MetaWRAP` is a very useful program with many tools for metagenome analysis but we will only use it to carry out functional annotations for now.

### Prokka: conda and directories {#prokka_conda}
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
```

Now to activate a `MetaWRAP` environment in a new terminal (right click on the main screen background, choose `Applications` -> `Shell` -> `bash`). The first step is to use `chos 8` which is a custom command for our cluster. It will Change the Unix OS to version 8 as version 7 (default in our cluster) is too old for some of the programs.

```{bash eval=FALSE}
chos 8
. usemetawrap
```

Make a new directory and move into it.

```{bash eval=FALSE}
mkdir ~/8-metawrap_anno
cd ~/8-metawrap_anno
```

### Prokka: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/enter_key.png", auto_pdf = TRUE)
``` 

Now we can annotate the bins.

The below will take a long time to run (>1 hour). Instead of running it skip onto the next step to copy pre-made output to continue with. This command is here so you know what to run in your own future analyses.

```{bash eval=FALSE}
metawrap annotate_bins \
-b ~/7-Binning/K1/bins \
-o ./K1 \
-t 8
```

The options used are:

- `-b`: The directory containing the assembled bins in fasta format.
- `-o`: The output directory.
- `-t`: Number of threads to use for the command.

### Prokka: output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/output.png", auto_pdf = TRUE)
``` 

Like in previous steps we will copy over premade results for the full dataset.

__Note__: This need to be done in one of you non `metawrap` terminals

```{bash eval=FALSE}
#Copy over directory
cp -r /pub39/tea/nsc006/NEOF/Shotgun_metagenomics/prokka/K1_fullset ~/8-metawrap_anno
#Change into copied directory
cd ~/8-metawrap_anno/K1_fullset
```

There are many directories but the one we are interested is `bin_funct_annotations` . This contains a GFF file for each bin.

Look at the gff file for bin 1.

```{bash eval=FALSE}
less bin_funct_annotations/K1.1.gff
```

The GFF file is a tab delimited file contain annotation information for the features in the assembly/bin. In this case it is a GFF3 file (most curent version of GFF).

There is quite a lot of information contained in each row so instead of listing all the columns here please have a look at the official documentation:

https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md

A quick thing we can do with these files is to see if any of the bins contain a specific annotation. For example, if we wanted to know if there were any ATP-binding proteins in any of the bins we could carry out the below command.

```{bash eval=FALSE}
grep "ATP-binding protein" bin_funct_annotations/K1*gff | less
```

We can now view the lines containing "ATP-binding protein" with the start of the line containing the file name the line belongs to.

What if you want to know about pathways?

## MinPath
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/pathways_2.png", auto_pdf = TRUE)
```

We can use `MinPath` to predict `MetaCyc` metabolic pathways. These pathway are made up of sets of enzymes that catalyse various reactions.

### MinPath: conda and directory
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
```

Use the `MetaWRAP` environment in `chos 8` for this section.

Ensure you start in the `~/8-metawrap_anno/K1_fullset` directory.

### MinPath: EC extraction
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/exctractor.png", auto_pdf = TRUE)
```

Before we can estimate the pathways we need to extract the EC numbers predictions from the GFF file. EC (Enzyme Commission) numbers are a numerical classification for enzymes based on the reaction they catalyse.

Unless you know the EC scheme well they are generally not helpful by themselves. An example EC number is EC 3.1.1.The numbers represents the group the enzyme belongs to with the first number being the biggest group. From highest to the lowest grouping 3.1.1. represents:

- __3__: Hydrolase enzymes.
- __3.1__: Hydrolase enzymes that act on ester bonds.
- __3.1.1__: Carboxylic Ester Hydrolases.

With all that information we will extract the EC annotations from the GFF files. First we'll create a directory for the output and move into the GFF directory.

```{bash eval=FALSE}
#Create EC directory
mkdir EC
#Change into GFF directory
cd bin_funct_annotations/
```

We will now use a loop with various parts to create an EC annotation file (`.ec`).

__Note__: The lack of `\` at the end of most of the ines is intentional. The below is all one command over multiple lines but loops work slightly different and don't need `\` in certain parts. Ensure you do press enter at the end of each line.

```{bash eval=FALSE}
ls -1 *gff | sed "s/.gff//" | while read bin
do
cat ${bin}.gff | grep "eC_number=" | cut -f 9 | cut -f 1,2 -d ';' | \
sed "s/ID=//g" | sed "s/;eC_number=/\t/g" > ../EC/${bin}.ec
done
```

That is quite a bit of code so we'll break it down with examples for you to run. Of course feel free to skip over the explanations of code sections you are already comfortable with.

The first part lists all our `.gff` files on one line each (`ls -1 *gff`). Then the suffix `.gff` is substituted with nothing `sed "s/.gff//"`. This gives us the name of each bin (e.g. `K1.1`, `K1.2`). Run this section of code to see that.

```{bash eval=FALSE}
#List all gff files on one (-1) each
ls -1 *gff
#List all the file prefixes (on one line each)
ls -1 *gff | sed "s/.gff//"
```

Next we loop through each file prefix and use the variable `bin` (arbitrarily chosen) which contains the file prefix. This is carried out with `while read bin`. All the lines between the `do` (start of the loop) and `done` (end of loop) are line run in the loop.

Run the loop with an echo command to show we are using the ${bin} variable to specify the input and output files.

```{bash eval=FALSE}
ls -1 *gff | sed "s/.gff//" | while read bin
do
echo "${bin}.gff ../EC/${bin}.ec"
done
```

Now to look at the command within the loop:

`cat ${bin}.gff | grep "eC_number=" | cut -f 9 | cut -f 1,2 -d ';' | \`
`sed "s/ID=//g" | sed "s/;eC_number=/\t/g" > ../EC/${bin}.ec`

A good way to figure out what a pipe workflow is doing is to to run the first part of workflow and then add section by section to see how it is affecting the output. Additionally, it is always good to run it on one file and `head` the output so we have a manageable amount of data to look at.

We'll do that with the `K1.1.gff` file.

__Note__: Remember we are adding `head` to the end for ease of viewing.

__Tips__: 

- Use the up arrow to go back to previously run commands that you can then edit.
- Remember the `clear` command.

```{bash eval=FALSE}
#Grab every line containing "ec_number="
cat K1.1.gff | grep "eC_number=" | head
#Cut out the 9th column/field (-f) (i.e. only keep the 9th column)
#cut uses tabs as the default column/field delimiter
cat K1.1.gff | grep "eC_number=" | cut -f 9 | head
#Cut out the 1st and 2nd field based on the delimeter ";" (-d ";")
cat K1.1.gff | grep "eC_number=" | cut -f 9 | cut -f 1,2 -d ';' | \
head
#Substitute (s/) ID= for nothing (/ID//)
#/g indicates to carry out it out for every instance in every line
#Default is to only the first instance in each line
cat K1.1.gff | grep "eC_number=" | cut -f 9 | cut -f 1,2 -d ';' | \
sed "s/ID=//g" | head
#Substitute ;eC_number= for a tab (\t)
cat K1.1.gff | grep "eC_number=" | cut -f 9 | cut -f 1,2 -d ';' | \
sed "s/ID=//g" | sed "s/;eC_number=/\t/g" | head
```

Finally the pipes output is written into a file: `> ../EC/${bin}.ec`.

The output `.ec` files are required for `MinPath` and are tab delimited with two columns:

1. protein/sequence id
2. annotation (in this case the EC number)

### MinPath: run
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/enter_key.png", auto_pdf = TRUE)
``` 

With our `.ec` files we can create our MetaCyc predictions.

First we'll change directory into the EC directory. Then create an output directory for the MetaCyc predictions.

```{bash eval=FALSE}
cd ../EC
mkdir ../MetaCyc
```

Now we can loop through the file suffixes to run MinPath.

```{bash eval=FALSE}
ls -1 *ec | sed "s/.ec//" | while read bin
do
python /pub39/tea/nsc006/programs/MinPath/MinPath.py \
-ec ${bin}.ec \
-report ../MetaCyc/${bin}.minpath
done
```

A lot of output will be printed to screen but this can be ignored unless you see warnings.

### MinPath: output
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/output.png", auto_pdf = TRUE)
``` 

First step is to change directory into the `MetaCyc` directory.

```{bash eval=FALSE}
#Change directory
cd ../MetaCyc
#List contents
ls
```

From the `CheckM` results we found that bin K1.22 was very good with a quality score >96%. We will therefore have a look at its output.

Have a look at the `.minpath` `-report` file for K1.22.

```{bash eval=FALSE}
less K1.22.minpath
```

The file contains the following columns

1. __Pathway ID__
2. __Pathway reconstruction__: Only available for genomes annotated in MetaCyc database.
3. __Naive__: Indicates if pathway was reconstructed by the naive mapping approach (1) or not (0).
4. __Minpath__: Indicates if the pathway was kept (1) or removed (0) by `MinPath`.
5. __Fam0__: The number of families involved in the pathway.
6. __Fam-found__: Number of families in pathway that were annotated/found.
7. __Name__: Description of pathway.

Quit (`q`) `less` when you are happy.

There are some quick things we can do in bash with these files.

```{bash eval=FALSE}
#Count number of pathways found in each bin with word count
wc -l *minpath
#Grab every line with "PWY-6972" from every file
grep "PWY-6972" *minpath | less
```

With this file you can then investigate what bins have which pathways. Additionally, with more samples analysed you can determine which samples have which pathways present.