<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Raw data | Shotgun Metagenomics</title>
  <meta name="description" content="Chapter 5 Raw data | Shotgun Metagenomics" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Raw data | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Raw data | Shotgun Metagenomics" />
  
  
  

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2021-05-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="04-Cluster_Introduction.html"/>
<link rel="next" href="06-Trimming_data.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-considerations"><i class="fa fa-check"></i><b>7.5</b> Host removal considerations</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-run"><i class="fa fa-check"></i><b>8.1.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-output"><i class="fa fa-check"></i><b>8.1.2</b> Kraken2: output</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.2</b> Krona plot</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.3</b> Bracken</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-run"><i class="fa fa-check"></i><b>8.3.1</b> Bracken: run</a></li>
<li class="chapter" data-level="8.3.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-output"><i class="fa fa-check"></i><b>8.3.2</b> Bracken: output</a></li>
<li class="chapter" data-level="8.3.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-merging-output"><i class="fa fa-check"></i><b>8.3.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="8.3.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-extracting-output"><i class="fa fa-check"></i><b>8.3.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.4</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-add-metadata"><i class="fa fa-check"></i><b>8.4.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="8.4.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-conda"><i class="fa fa-check"></i><b>8.4.2</b> LEfSe: conda</a></li>
<li class="chapter" data-level="8.4.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-format"><i class="fa fa-check"></i><b>8.4.3</b> LEfSe: format</a></li>
<li class="chapter" data-level="8.4.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-run"><i class="fa fa-check"></i><b>8.4.4</b> LEfSe: run</a></li>
<li class="chapter" data-level="8.4.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-visualisation"><i class="fa fa-check"></i><b>8.4.5</b> LEfSe: visualisation</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>8.5</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2"><i class="fa fa-check"></i><b>9.1</b> HUMAnN2</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-conda-directories-and-files"><i class="fa fa-check"></i><b>9.1.1</b> HUMAnN2: conda, directories, and files</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-run"><i class="fa fa-check"></i><b>9.1.2</b> HUMAnN2: run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-output"><i class="fa fa-check"></i><b>9.1.3</b> HUMAnN2: output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#statistical-comparison-between-samples"><i class="fa fa-check"></i><b>9.2</b> Statistical comparison between samples</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-combining-data"><i class="fa fa-check"></i><b>9.2.1</b> HUMAnN2 comparison: combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-renormalising-data"><i class="fa fa-check"></i><b>9.2.2</b> HUMAnN2 comparison: renormalising data</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-pca-plot"><i class="fa fa-check"></i><b>9.2.3</b> HUMAnN2 comparison: PCA plot</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-lefse"><i class="fa fa-check"></i><b>9.2.4</b> HUMAnN2 comparison: LEfSe</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.1</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.2</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.3</b> Assembly</a></li>
<li class="chapter" data-level="10.4" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.4</b> QUAST</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html"><i class="fa fa-check"></i><b>11</b> Genome binning</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>11.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-conda-environment-directory"><i class="fa fa-check"></i><b>11.1.1</b> MetaBAT2: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.1.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>11.1.2</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="11.1.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>11.1.3</b> MetaBAT2: run</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>11.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-conda"><i class="fa fa-check"></i><b>11.2.1</b> CheckM: Conda</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>11.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="11.2.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>11.2.3</b> CheckM: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-MetaWrap.html"><a href="12-MetaWrap.html"><i class="fa fa-check"></i><b>12</b> MetaWRAP</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-MetaWrap.html"><a href="12-MetaWrap.html#bin-refinement"><i class="fa fa-check"></i><b>12.1</b> Bin refinement</a></li>
<li class="chapter" data-level="12.2" data-path="12-MetaWrap.html"><a href="12-MetaWrap.html#bin-reassembly"><i class="fa fa-check"></i><b>12.2</b> Bin reassembly</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="16-Appendix.html"><a href="16-Appendix.html"><i class="fa fa-check"></i><b>13</b> Appendix</a>
<ul>
<li class="chapter" data-level="13.1" data-path="16-Appendix.html"><a href="16-Appendix.html#manuals"><i class="fa fa-check"></i><b>13.1</b> Manuals</a></li>
<li class="chapter" data-level="13.2" data-path="16-Appendix.html"><a href="16-Appendix.html#cog-cateories"><i class="fa fa-check"></i><b>13.2</b> COG Cateories</a></li>
<li class="chapter" data-level="13.3" data-path="16-Appendix.html"><a href="16-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>13.3</b> Obtaining Read Data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="raw-data" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Raw data</h1>
<p><img src="figures/usb_stick.png" width="20%" style="display: block; margin: auto;" /></p>
<p>The very first thing we need to do is to obtain a dataset to work with. The European Bioinformatics Institute (EBI) provides an excellent metagenomics resource (<a href="https://www.ebi.ac.uk/metagenomics/" class="uri">https://www.ebi.ac.uk/metagenomics/</a>) which allows users to download publicly available metagenomic and metagenetic datasets.</p>
<p>Have a browse of some of the projects by selecting one of the biomes on this page.</p>
<p>We have selected a dataset from this site that consists of DNA shotgun data generated from 24 human faecal samples. 12 of these samples are from subjects who were fed a western diet and 12 are from subjects who were fed a Korean diet. This dataset comes from the EBI metagenomics resource (<a href="https://www.ebi.ac.uk/metagenomics/projects/ERP005558" class="uri">https://www.ebi.ac.uk/metagenomics/projects/ERP005558</a>).</p>
<div id="obtaining-the-data" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Obtaining the data</h2>
<p><img src="figures/link.png" width="15%" style="display: block; margin: auto;" /></p>
<p>First, we need to create a directory to put the data in and then change directory to it.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="05-Raw_data.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 1-Raw</span>
<span id="cb3-2"><a href="05-Raw_data.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 1-Raw</span></code></pre></div>
<p>Now we can generate a symbolic links (i.e. shortcut) to the raw sequence data files, which will appear in the current directory:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="05-Raw_data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> -s /pub39/tea/matthew/NEOF/Shotgun_metagenomics/raw_fastq/* .</span></code></pre></div>
<p>If you would like to know more about the <code>ln</code> command please check out: <a href="https://linuxize.com/post/how-to-create-symbolic-links-in-linux-using-the-ln-command/" class="uri">https://linuxize.com/post/how-to-create-symbolic-links-in-linux-using-the-ln-command/</a>.</p>
<p>Now, check the symbolic links are in your current directory:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="05-Raw_data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>There should be six files in the directory, two for each sample in the dataset. e.g. <code>K1_R1.fastq.gz</code></p>
<p>The file ID has three components:</p>
<ul>
<li>K1 is the sample ID.</li>
<li>R1 is for the forward reads in the Illumina reads pair (R2 is for the set corresponding to the other end of the reads).</li>
<li>fastq.gz tells us that this is a zipped FASTQ file.</li>
</ul>
<p>The sample labelling indicates the type treatment samples. The three samples are:</p>
<ul>
<li><strong>K1</strong>: Fecal sample of individual of Korean diets</li>
<li><strong>K2</strong>: Fecal sample of individual of Korean diets</li>
<li><strong>W1</strong>: Fecal sample of individual of Western diets</li>
</ul>
<p>So, what do the R1 and R2 actually mean? With Illumina sequencing the vast majority of sequencing is paired end. i.e. DNA is first fragmented and both ends of each fragment are sequenced as shown here:</p>
<p><img src="figures/paired_reads.png" width="40%" style="display: block; margin: auto;" /></p>
<p>This results in two sequences generated for each sequenced fragment: One reading in from the 3’ end (R1) and the other reading in from the 5’ end (R2).</p>
<p>FASTQ is a sequence format much like FASTA, with the addition of quality scores. To see what a FASTQ file looks like, we can inspect the first few lines on one of our sequence files:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="05-Raw_data.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">zcat</span> K1_R1.fastq.gz <span class="kw">|</span> <span class="fu">head</span> -n 4 <span class="kw">|</span> <span class="fu">less</span> -S</span></code></pre></div>
<p>The pipe symbol ( <code>|</code> ) is used to pass the output of one command as input to the next command. So, this command (1) shows the unzipped contents of the FASTQ file, (2) displays only the first 4 lines, and (3) displays them without wrapping lines (with <code>–S</code>, for easy viewing).</p>
<p>The lines displayed represent one FASTQ sequence entry, or one read of a read pair: The corresponding second read can be viewed by running the same command on K1_R2.fastq.gz. The first line is the read identifier, the second line is the sequence itself, the third line is a secondary header (which is usually left blank except for ‘+’) and the fourth line is the sequence quality score: For each base in the sequence, there is a corresponding quality encoded in this string of characters. <strong>To return to the command prompt, press</strong> <code>q</code>.</p>
<p>Due to computational constraints, the files you have linked to are a subset of the original data (i.e. 1 million read pairs from each sample).</p>
</div>
<div id="checking-quality-control" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Checking quality control</h2>
<p><img src="figures/mangify_glass.png" width="15%" style="display: block; margin: auto;" /></p>
<p>We can generate and visualise various sequence data metrics for quality control purposes using <code>FastQC</code>. We will run <code>FastQC</code> on the R1 and R2 reads separately as it is good to visualise them in two different reports. This is beacause R1 and R2 reads have different quality patterns, generally due to the poorer quality of R2.</p>
<p>Run <code>FastQC</code> on the files:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="05-Raw_data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#R1 fastqc</span></span>
<span id="cb7-2"><a href="05-Raw_data.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Make an output directory</span></span>
<span id="cb7-3"><a href="05-Raw_data.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> R1_fastq</span>
<span id="cb7-4"><a href="05-Raw_data.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Run fastqc on all the R1.fastq.gz files </span></span>
<span id="cb7-5"><a href="05-Raw_data.html#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#* matches any pattern</span></span>
<span id="cb7-6"><a href="05-Raw_data.html#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#*R1.fastq.gz matches any file that ends R1.fastq.gz in the current directory</span></span>
<span id="cb7-7"><a href="05-Raw_data.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#-t 3 indicates to use 3 threads, chosen as there are three R1 files</span></span>
<span id="cb7-8"><a href="05-Raw_data.html#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> -t 3 -o R1_fastqc *R1.fastq.gz</span>
<span id="cb7-9"><a href="05-Raw_data.html#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="05-Raw_data.html#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#R2 fastqc</span></span>
<span id="cb7-11"><a href="05-Raw_data.html#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Make output directory</span></span>
<span id="cb7-12"><a href="05-Raw_data.html#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> R2_fastq</span>
<span id="cb7-13"><a href="05-Raw_data.html#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Run fastqc</span></span>
<span id="cb7-14"><a href="05-Raw_data.html#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> -t 3 -o R2_fastqc *R2.fastq.gz</span></code></pre></div>
<p>Once the <code>FastQC</code> commands are run we can run <code>MultiQC</code> to create interactive html reports for the outputs.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="05-Raw_data.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#R1 multiqc fastqc report</span></span>
<span id="cb8-2"><a href="05-Raw_data.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Create output directory</span></span>
<span id="cb8-3"><a href="05-Raw_data.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> R1_fastqc/multiqc</span>
<span id="cb8-4"><a href="05-Raw_data.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Create multiqc output</span></span>
<span id="cb8-5"><a href="05-Raw_data.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> -o R1_fastqc/multiqc R1_fastqc</span>
<span id="cb8-6"><a href="05-Raw_data.html#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="05-Raw_data.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#R2 multiqc fastqc report</span></span>
<span id="cb8-8"><a href="05-Raw_data.html#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Create output directory</span></span>
<span id="cb8-9"><a href="05-Raw_data.html#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> R2_fastqc/multiqc</span>
<span id="cb8-10"><a href="05-Raw_data.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Create multiqc report</span></span>
<span id="cb8-11"><a href="05-Raw_data.html#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> -o R2_fastqc/multiqc R2_fastqc</span></code></pre></div>
<p>Once completed, view the <code>MultiQC</code> reports (NB: The <code>&amp;</code> runs the command in the background, therefore allowing you to continue to run commands while Firefox is still open):</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="05-Raw_data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> R1_fastqc/multiqc/multiqc_report.html <span class="kw">\</span></span>
<span id="cb9-2"><a href="05-Raw_data.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">R2_fastqc/multiqc/multiqc_report.html</span> <span class="kw">&amp;</span></span></code></pre></div>
<p>The <code>FastQC</code> report (via <code>MultiQC</code>) contains a number of metrics. The “Sequence Quality Histograms” shows the sequence quality across the length of the reads, you can hover over each line to show which sample it belongs to. Note how quality decreases as the length of the read increases. While this is normal with Illumina sequencing, we will improve the situation a bit in the next chapter.</p>
<p>Once you have finished inspecting, minimise the Firefox window.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="04-Cluster_Introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="06-Trimming_data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
