<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Quality control | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Quality control | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Quality control | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2022-04-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-Raw_data.html"/>
<link rel="next" href="07-Host_removal.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-considerations"><i class="fa fa-check"></i><b>7.5</b> Host removal considerations</a></li>
</ul></li>
<li class="part"><span><b>Read approach</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-run"><i class="fa fa-check"></i><b>8.1.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-output"><i class="fa fa-check"></i><b>8.1.2</b> Kraken2: output</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.2</b> Krona plot</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.3</b> Bracken</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-run"><i class="fa fa-check"></i><b>8.3.1</b> Bracken: run</a></li>
<li class="chapter" data-level="8.3.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-output"><i class="fa fa-check"></i><b>8.3.2</b> Bracken: output</a></li>
<li class="chapter" data-level="8.3.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-merging-output"><i class="fa fa-check"></i><b>8.3.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="8.3.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-extracting-output"><i class="fa fa-check"></i><b>8.3.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.4</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2_lefse_metadata"><i class="fa fa-check"></i><b>8.4.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="8.4.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-conda"><i class="fa fa-check"></i><b>8.4.2</b> LEfSe: conda</a></li>
<li class="chapter" data-level="8.4.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-format"><i class="fa fa-check"></i><b>8.4.3</b> LEfSe: format</a></li>
<li class="chapter" data-level="8.4.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-run"><i class="fa fa-check"></i><b>8.4.4</b> LEfSe: run</a></li>
<li class="chapter" data-level="8.4.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-visualisation"><i class="fa fa-check"></i><b>8.4.5</b> LEfSe: visualisation</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>8.5</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> HUMAnN Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann"><i class="fa fa-check"></i><b>9.1</b> HUMAnN</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann-conda-directories-and-files"><i class="fa fa-check"></i><b>9.1.1</b> HUMAnN: conda, directories, and files</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann-run"><i class="fa fa-check"></i><b>9.1.2</b> HUMAnN: run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann-output"><i class="fa fa-check"></i><b>9.1.3</b> HUMAnN: output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humannmultisampleprocessing"><i class="fa fa-check"></i><b>9.2</b> Multi sample processing</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann-combining-data"><i class="fa fa-check"></i><b>9.2.1</b> HUMAnN: combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann-split-stratified-table"><i class="fa fa-check"></i><b>9.2.2</b> HUMAnN: split stratified table</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humannrenormalise"><i class="fa fa-check"></i><b>9.2.3</b> HUMAnN: renormalising data</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#heatmap"><i class="fa fa-check"></i><b>9.3</b> Heatmap</a></li>
<li class="chapter" data-level="9.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann_lefse"><i class="fa fa-check"></i><b>9.4</b> LEfSe</a></li>
<li class="chapter" data-level="9.5" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#optional-task"><i class="fa fa-check"></i><b>9.5</b> Optional task</a></li>
</ul></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#metagenome-assembly-conda"><i class="fa fa-check"></i><b>10.1</b> Metagenome assembly: Conda</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.2</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.3</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.4" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.4</b> Assembly</a></li>
<li class="chapter" data-level="10.5" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.5</b> QUAST</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html"><i class="fa fa-check"></i><b>11</b> Genome binning</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>11.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-conda-environment-directory"><i class="fa fa-check"></i><b>11.1.1</b> MetaBAT2: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.1.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>11.1.2</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="11.1.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>11.1.3</b> MetaBAT2: run</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>11.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-conda"><i class="fa fa-check"></i><b>11.2.1</b> CheckM: Conda</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>11.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="11.2.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>11.2.3</b> CheckM: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>12</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>12.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="12.2" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka"><i class="fa fa-check"></i><b>12.2</b> Prokka</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka_conda"><i class="fa fa-check"></i><b>12.2.1</b> Prokka: conda and directories</a></li>
<li class="chapter" data-level="12.2.2" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka-run"><i class="fa fa-check"></i><b>12.2.2</b> Prokka: run</a></li>
<li class="chapter" data-level="12.2.3" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka-output"><i class="fa fa-check"></i><b>12.2.3</b> Prokka: output</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>12.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-conda-and-directory"><i class="fa fa-check"></i><b>12.3.1</b> MinPath: conda and directory</a></li>
<li class="chapter" data-level="12.3.2" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>12.3.2</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-run"><i class="fa fa-check"></i><b>12.3.3</b> MinPath: run</a></li>
<li class="chapter" data-level="12.3.4" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>12.3.4</b> MinPath: output</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="13-Appendix.html"><a href="13-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="13-Appendix.html"><a href="13-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="13-Appendix.html"><a href="13-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qualcont" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Quality control</h1>
<p><img src="figures/quality_trimming_and_filtering.png" width="30%" style="display: block; margin: auto;" /></p>
<p>Now that we’ve obtained the raw data and had a look at it, we should now clean it up. With any sequencing data, it is very important to ensure that you use the highest quality data possible: Rubbish goes in, rubbish comes out.</p>
<p>There are two main methods employed to clean sequence data, and a third method specific to some metagenomic datasets.</p>
<ul>
<li>Remove low quality bases from the end of the reads: These are more likely to be incorrect, so are best trimmed off.</li>
<li>Remove adapters: Sometimes sequencing adapters can be sequenced if the sequencing runs off the end of a fragment.</li>
<li>Host removal: If a metagenomic sample derives from a host species then it may be advisable to remove any reads associated with the host genome. Here, we do not need to do this, as the dataset has next to zero human genome sequences.</li>
</ul>
<div id="removing-adapters-and-low-quality-bases" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Removing adapters and low quality bases</h2>
<p><img src="figures/hedge_trimming.png" width="15%" style="display: block; margin: auto;" /></p>
<p>First go back to your home directory and create a new directory where we will clean the sequences up:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="06-Trimming_data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb10-2"><a href="06-Trimming_data.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 2-Trimmed</span>
<span id="cb10-3"><a href="06-Trimming_data.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 2-Trimmed</span></code></pre></div>
<p>You are now in your newly created directory. Here we will run <code>Trim Galore!</code> which removes low quality bases and adapters.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="06-Trimming_data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">trim_galore</span> <span class="at">--paired</span> <span class="at">--quality</span> 20 <span class="at">--stringency</span> 4 <span class="dt">\</span></span>
<span id="cb11-2"><a href="06-Trimming_data.html#cb11-2" aria-hidden="true" tabindex="-1"></a>../1-Raw/K1_R1.fastq.gz ../1-Raw/K1_R2.fastq.gz</span></code></pre></div>
<p>This is a longer command so we’ve split it across multiple lines. A <code>\</code> at the end of a line allows you to press return without running the command, meaning you can continue to add to that command. When this happens, the <code>$</code> changes to a <code>&gt;</code>.</p>
<p><strong>Note</strong> if you do use the <code>\</code> character, the next key you press must be return. If you use <code>\</code> in the middle of a line without pressing return afterwards, it will break the command!</p>
<p>This command will remove any low quality regions from the end of both reads in each read pair (quality score &lt; 20). Additionally, if it detects four or more bases of a sequencing adapter, it will trim that off too.</p>
<p><strong>Task</strong>: Rerun this command for the other two samples (K2 and W1).</p>
</div>
<div id="rename-the-files" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Rename the files</h2>
<p><img src="figures/rename.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Once that is complete if you run:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="06-Trimming_data.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>you will notice that we have a new bunch of files created: 2 new read files for each sample along with a trimming report for each file trimmed. However, the new names are needlessly long. For example K1_R1_val_R1.fq.gz could be shortened to K1_R1.fq.gz. So, we’ll rename all of the files with the mv command:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="06-Trimming_data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> K1_R1_val_1.fq.gz K1_R1.fq.gz</span>
<span id="cb13-2"><a href="06-Trimming_data.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> K1_R2_val_2.fq.gz K1_R2.fq.gz</span>
<span id="cb13-3"><a href="06-Trimming_data.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> K2_R1_val_1.fq.gz K2_R1.fq.gz</span>
<span id="cb13-4"><a href="06-Trimming_data.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> K2_R2_val_2.fq.gz K2_R2.fq.gz</span>
<span id="cb13-5"><a href="06-Trimming_data.html#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> W1_R1_val_1.fq.gz W1_R1.fq.gz</span>
<span id="cb13-6"><a href="06-Trimming_data.html#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> W1_R2_val_2.fq.gz W1_R2.fq.gz</span></code></pre></div>
<p><strong>Tip</strong>: If you want to edit and reuse previous commands, press the up arrow key.</p>
<p><strong>Task</strong>: Briefly inspect the log files to see how the trimming went (e.g. K1_R1.fastq.gz_trimming_report.txt).</p>
</div>
<div id="inspect-the-trimmed-data" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Inspect the trimmed data</h2>
<p><img src="figures/magnify_glass_good.png" width="15%" style="display: block; margin: auto;" /></p>
<p>To see what difference the trimming made, run <code>FastQC</code> and <code>MultiQC</code> again on the trimmed output files and view it.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="06-Trimming_data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#R1 fastqc and multiqc</span></span>
<span id="cb14-2"><a href="06-Trimming_data.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> R1_fastqc</span>
<span id="cb14-3"><a href="06-Trimming_data.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> <span class="at">-t</span> 3 <span class="at">-o</span> R1_fastqc <span class="pp">*</span>R1.fq.gz</span>
<span id="cb14-4"><a href="06-Trimming_data.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> R1_fastqc/multiqc</span>
<span id="cb14-5"><a href="06-Trimming_data.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> <span class="at">-o</span> R1_fastqc/multiqc R1_fastqc</span></code></pre></div>
<p><strong>Task</strong>: Run <code>FastQC</code> and <code>MultiQC</code> for the R2 files and then view the R1 and R2 <code>MultiQC</code> reports with firefox. How does the quality compare to the untrimmed data?</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="05-Raw_data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-Host_removal.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
