<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Genome binning | Shotgun Metagenomics</title>
  <meta name="description" content="Chapter 11 Genome binning | Shotgun Metagenomics" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Genome binning | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Genome binning | Shotgun Metagenomics" />
  
  
  

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2021-05-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="10-Metagenome_assembly.html"/>
<link rel="next" href="12-MAG_annotation.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-considerations"><i class="fa fa-check"></i><b>7.5</b> Host removal considerations</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-run"><i class="fa fa-check"></i><b>8.1.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-output"><i class="fa fa-check"></i><b>8.1.2</b> Kraken2: output</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.2</b> Krona plot</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.3</b> Bracken</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-run"><i class="fa fa-check"></i><b>8.3.1</b> Bracken: run</a></li>
<li class="chapter" data-level="8.3.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-output"><i class="fa fa-check"></i><b>8.3.2</b> Bracken: output</a></li>
<li class="chapter" data-level="8.3.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-merging-output"><i class="fa fa-check"></i><b>8.3.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="8.3.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-extracting-output"><i class="fa fa-check"></i><b>8.3.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.4</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-add-metadata"><i class="fa fa-check"></i><b>8.4.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="8.4.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-conda"><i class="fa fa-check"></i><b>8.4.2</b> LEfSe: conda</a></li>
<li class="chapter" data-level="8.4.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-format"><i class="fa fa-check"></i><b>8.4.3</b> LEfSe: format</a></li>
<li class="chapter" data-level="8.4.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-run"><i class="fa fa-check"></i><b>8.4.4</b> LEfSe: run</a></li>
<li class="chapter" data-level="8.4.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-visualisation"><i class="fa fa-check"></i><b>8.4.5</b> LEfSe: visualisation</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>8.5</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2"><i class="fa fa-check"></i><b>9.1</b> HUMAnN2</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-conda-directories-and-files"><i class="fa fa-check"></i><b>9.1.1</b> HUMAnN2: conda, directories, and files</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-run"><i class="fa fa-check"></i><b>9.1.2</b> HUMAnN2: run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-output"><i class="fa fa-check"></i><b>9.1.3</b> HUMAnN2: output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#statistical-comparison-between-samples"><i class="fa fa-check"></i><b>9.2</b> Statistical comparison between samples</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-combining-data"><i class="fa fa-check"></i><b>9.2.1</b> HUMAnN2 comparison: combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-renormalising-data"><i class="fa fa-check"></i><b>9.2.2</b> HUMAnN2 comparison: renormalising data</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-pca-plot"><i class="fa fa-check"></i><b>9.2.3</b> HUMAnN2 comparison: PCA plot</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-lefse"><i class="fa fa-check"></i><b>9.2.4</b> HUMAnN2 comparison: LEfSe</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.1</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.2</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.3</b> Assembly</a></li>
<li class="chapter" data-level="10.4" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.4</b> QUAST</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html"><i class="fa fa-check"></i><b>11</b> Genome binning</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>11.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-conda-environment-directory"><i class="fa fa-check"></i><b>11.1.1</b> MetaBAT2: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.1.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>11.1.2</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="11.1.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>11.1.3</b> MetaBAT2: run</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>11.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-conda"><i class="fa fa-check"></i><b>11.2.1</b> CheckM: Conda</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>11.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="11.2.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>11.2.3</b> CheckM: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-MAG_annotation.html"><a href="12-MAG_annotation.html"><i class="fa fa-check"></i><b>12</b> MAG annotation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-MAG_annotation.html"><a href="12-MAG_annotation.html#phylophlan3"><i class="fa fa-check"></i><b>12.1</b> PhyloPhlAn3</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="12-MAG_annotation.html"><a href="12-MAG_annotation.html#phylophlan3-conda-and-directories"><i class="fa fa-check"></i><b>12.1.1</b> PhyloPhlAn3: conda and directories</a></li>
<li class="chapter" data-level="12.1.2" data-path="12-MAG_annotation.html"><a href="12-MAG_annotation.html#phylophlan3-configuration-files"><i class="fa fa-check"></i><b>12.1.2</b> PhyloPhlAn3: configuration files</a></li>
<li class="chapter" data-level="12.1.3" data-path="12-MAG_annotation.html"><a href="12-MAG_annotation.html#phylophlan3-run"><i class="fa fa-check"></i><b>12.1.3</b> PhyloPhlAn3: run</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-MetaWrap.html"><a href="13-MetaWrap.html"><i class="fa fa-check"></i><b>13</b> MetaWRAP</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-MetaWrap.html"><a href="13-MetaWrap.html#tutorial"><i class="fa fa-check"></i><b>13.1</b> Tutorial</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-Appendix.html"><a href="14-Appendix.html"><i class="fa fa-check"></i><b>14</b> Appendix</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Appendix.html"><a href="14-Appendix.html#next-steps"><i class="fa fa-check"></i><b>14.1</b> Next steps</a></li>
<li class="chapter" data-level="14.2" data-path="14-Appendix.html"><a href="14-Appendix.html#visualising-data"><i class="fa fa-check"></i><b>14.2</b> Visualising data</a></li>
<li class="chapter" data-level="14.3" data-path="14-Appendix.html"><a href="14-Appendix.html#manuals"><i class="fa fa-check"></i><b>14.3</b> Manuals</a></li>
<li class="chapter" data-level="14.4" data-path="14-Appendix.html"><a href="14-Appendix.html#cog-cateories"><i class="fa fa-check"></i><b>14.4</b> COG Cateories</a></li>
<li class="chapter" data-level="14.5" data-path="14-Appendix.html"><a href="14-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>14.5</b> Obtaining Read Data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genome-binning" class="section level1" number="11">
<h1><span class="header-section-number">Chapter 11</span> Genome binning</h1>
<p><img src="figures/yarn_binning.png" width="40%" style="display: block; margin: auto;" /></p>
<p>A metagenome assembly consists of contigs from many different genomes. At this stage we don’t know which contigs are from which species. We could try to taxonomically classify each contig but there are 2 problems with this approach:</p>
<ol style="list-style-type: decimal">
<li>Some contigs may be misclassified which can lead to multiple contigs from the same genome/organism being classified as various taxa.</li>
<li>Databases are incomplete and so some contigs will not be classified at all (microbial dark matter).</li>
</ol>
<p>To alleviate these issues genomic binning can be carried out. This will cluster contigs into bins based on:</p>
<ul>
<li><strong>Coverage</strong>: Contigs with similar coverage are more likely to be from the same genome.</li>
<li><strong>Composition</strong>: Contigs with similar GC content are more likely to belong to the same genome.</li>
</ul>
<p>Genomic binning has been used to discover many new genomes. Additionally, it makes downstream analyses quicker as the downstream steps will be carried out on the sets of bins rather than on one large metagenome assembly.</p>
<p>Binning produces “bins” of contigs of various quality (e.g. draft, complete). These bins are also know as MAGs (Metagenome-assembled genomes). In other words a MAG is a single assembled genome that was assembled with other genomes in a metagenome assembly but later separated from the other assemblies. The term MAG has been adopted by the GSC (Genomics Standards Consortium).</p>
<p>It is recommended to ensure you do not have a poor quality metagenome assembly. Binning requires contigs of good length and good coverage. Extremely low coverage and very short contigs will be excluded from binning.</p>
<div id="metabat2" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> MetaBAT2</h2>
<p><img src="figures/bat.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use <code>MetaBAT2</code> for our genome binning. It is a relatively new binning tool with three major upsides that makes it very popular:</p>
<ol style="list-style-type: decimal">
<li>It has very reliable default parameters meaning virtually no parameter optimisation is required.</li>
<li>It performs very well amongst genome binners.</li>
<li>It is computationally efficient compared to other binners (requires less RAM, cores etc.)</li>
</ol>
<div id="metabat2-conda-environment-directory" class="section level3" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> MetaBAT2: Conda environment &amp; directory</h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>shotgun_meta</code> conda environment to start for this chapter.</p>
<p>Additionally, make a new directory and move into it.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb61-1"><a href="11-Genome_binning.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make directory</span></span>
<span id="cb61-2"><a href="11-Genome_binning.html#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> -p ~/7-Binning/K1</span>
<span id="cb61-3"><a href="11-Genome_binning.html#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Move into it</span></span>
<span id="cb61-4"><a href="11-Genome_binning.html#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/7-Binning/K1</span></code></pre></div>
</div>
<div id="metabat2-depth-calculation" class="section level3" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> MetaBAT2: depth calculation</h3>
<p><img src="figures/submarine.png" width="10%" style="display: block; margin: auto;" /></p>
<p>To carry out effective genome binning <code>MetaBAT2</code> uses coverage information of the contigs. To calculate depth we need to align the reads to the metagenome assembly.</p>
<p>For the alignment we will use <code>bwa</code>. We need to index our assembly file prior to alignment.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="11-Genome_binning.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> index ~/6-Assembly/K1/final.contigs.fa</span></code></pre></div>
<p>Next we will align our trimmed paired reads we used to create the stitched reads. We will carry this out with the <code>bwa mem</code> command. <code>bwa mem</code> is a good aligner for short reads. If you are using long reads (PacBio or Nanopore) <code>minimap2</code> will be more appropriate.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb63-1"><a href="11-Genome_binning.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem ~/6-Assembly/K1/final.contigs.fa <span class="kw">\</span></span>
<span id="cb63-2"><a href="11-Genome_binning.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="ex">~/2-Trimmed/K1_R1.final.fastq</span> ~/2-Trimmed/K1_R2.final.fastq <span class="op">&gt;</span> <span class="kw">\</span></span>
<span id="cb63-3"><a href="11-Genome_binning.html#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="ex">K1.sam</span></span></code></pre></div>
<p>After alignment we need to get the file ready for the contig depth summarisation step. This requires converting the <code>sam</code> file to a <code>bam</code> (binary form of a <code>sam</code> file) file and then sorting the <code>bam</code> file.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="11-Genome_binning.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert sam to bam file</span></span>
<span id="cb64-2"><a href="11-Genome_binning.html#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view -bu K1.sam <span class="op">&gt;</span> K1.bam</span>
<span id="cb64-3"><a href="11-Genome_binning.html#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Created sorted bam file</span></span>
<span id="cb64-4"><a href="11-Genome_binning.html#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort K1.bam <span class="op">&gt;</span> K1.sort.bam</span></code></pre></div>
<p>Now we can summarise the contig depths from the sorted <code>bam</code> files with <code>MetaBAT2</code>’s <code>jgi_summarize_bam_contig_depths</code> command.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="11-Genome_binning.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="ex">jgi_summarize_bam_contig_depths</span> --outputDepth K1.depth.txt K1.sort.bam</span></code></pre></div>
<p>You can have a look at the depth file and you will notice there are many contigs with low coverage (&lt;10) and of short length (&lt;1500).</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb66-1"><a href="11-Genome_binning.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> K1.depth.txt</span></code></pre></div>
<p>To get a better look we will open the file in <code>R</code> and look at a summary of the file’s table.</p>
<p>Activate <code>R</code>:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb67-1"><a href="11-Genome_binning.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>Now in <code>R</code> we will read in the file and get a <code>summary()</code> of it.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="11-Genome_binning.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in the table as an object called df (short for data frame)</span></span>
<span id="cb68-2"><a href="11-Genome_binning.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="co">#We want the first row to be the column names (header=TRUE)</span></span>
<span id="cb68-3"><a href="11-Genome_binning.html#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="co">#We do not want R to check the column names and &quot;fix&quot; them (check.names=FALSE)</span></span>
<span id="cb68-4"><a href="11-Genome_binning.html#cb68-4" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;K1.depth.txt&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb68-5"><a href="11-Genome_binning.html#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Create a summary of the data</span></span>
<span id="cb68-6"><a href="11-Genome_binning.html#cb68-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(df)</span></code></pre></div>
<p>We can see the numbers of the <code>contigLen</code> and <code>totalAvgDepth</code> are very low. However, this is most likely due to a bunch of short low coverage contigs which will be ignored by <code>MetaBAT2</code>. Therefore we will remove rows with information on contigs shorter than 1500 and rerun the summary. <code>MetaBAT2</code>’s documentation dictates the minimum contig length should be &gt;=1500 with its default being 2500.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="11-Genome_binning.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Set the new object &quot;df_min1500len&quot; as all rows</span></span>
<span id="cb69-2"><a href="11-Genome_binning.html#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="co">#where the value in the column &quot;contigLen&quot; of &quot;df&quot;</span></span>
<span id="cb69-3"><a href="11-Genome_binning.html#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Is greater than or equal to 1500</span></span>
<span id="cb69-4"><a href="11-Genome_binning.html#cb69-4" aria-hidden="true" tabindex="-1"></a>df_min1500len <span class="ot">&lt;-</span> df[df<span class="sc">$</span>contigLen <span class="sc">&gt;=</span> <span class="dv">1500</span>,]</span>
<span id="cb69-5"><a href="11-Genome_binning.html#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Summary of our new data frame</span></span>
<span id="cb69-6"><a href="11-Genome_binning.html#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(df_min1500len)</span></code></pre></div>
<p>That is looking better. The minimum average coverage for <code>MetaBAT2</code> is 1 and our minimum is higher than that. Now you can quit R and continue.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="11-Genome_binning.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co">#quit R</span></span>
<span id="cb70-2"><a href="11-Genome_binning.html#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">q</span>()</span>
<span id="cb70-3"><a href="11-Genome_binning.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="co">#On the prompt to save your workspace press &quot;n&quot; and then enter.</span></span></code></pre></div>
<p><strong>Note</strong>: One of the reasons for our short contigs is that we only used a subset of our sequencing dataset for this tutorial due to time concerns.</p>
</div>
<div id="metabat2-run" class="section level3" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> MetaBAT2: run</h3>
<p><img src="figures/enter_key.png" width="10%" style="display: block; margin: auto;" /></p>
<p>With our assembly and its depth information we can run <code>MetaBAT2</code> for binning.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb71-1"><a href="11-Genome_binning.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a diretcory for the bins</span></span>
<span id="cb71-2"><a href="11-Genome_binning.html#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> bins</span>
<span id="cb71-3"><a href="11-Genome_binning.html#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Run MetaBAT2</span></span>
<span id="cb71-4"><a href="11-Genome_binning.html#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="ex">metabat2</span> <span class="kw">\</span></span>
<span id="cb71-5"><a href="11-Genome_binning.html#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="ex">--inFile</span> ~/6-Assembly/K1/final.contigs.fa <span class="kw">\</span></span>
<span id="cb71-6"><a href="11-Genome_binning.html#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="ex">--outFile</span> bins/K1 <span class="kw">\</span></span>
<span id="cb71-7"><a href="11-Genome_binning.html#cb71-7" aria-hidden="true" tabindex="-1"></a><span class="ex">--abdFile</span> K1.depth.txt <span class="kw">\</span></span>
<span id="cb71-8"><a href="11-Genome_binning.html#cb71-8" aria-hidden="true" tabindex="-1"></a><span class="ex">--minContig</span> 1500</span></code></pre></div>
<p>List the contents of the output directory and you’ll see there are 2 files with the prefix of <code>K1</code>. These are the 2 bins that will hopefully contain 1 MAG (Metagenome-Assembled Genome).</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="11-Genome_binning.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> bins</span></code></pre></div>
</div>
</div>
<div id="checkm" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> CheckM</h2>
<p><img src="figures/CheckM.png" width="40%" style="display: block; margin: auto;" /></p>
<p><code>CheckM</code> is a useful tool to assess the quality of assembled bacterial and archaeal genomes. This can be used on assemblies produced from single cell, single isolate, or metagenome data. Additionally, it can be used to identify bins that are likely candidates for merging. This occurs when one genome has been separated into different bins.</p>
<p>An important part of CheckM is the ubiquitous and single-copy genes it utilises. It has sets of these genes for different phylogenetic lineages. With these it can determine:</p>
<ul>
<li>What lineage a bin/MAG belongs to.
<ul>
<li>Does it contain genes only found in <em>Escherichia</em>?</li>
</ul></li>
<li>How complete the bin/MAG is. A set of lineage specific genes should all be found in a genome belonging to the lineage (ubiquitous).
<ul>
<li>What percentage of these lineage specific genes are present in the MAG?</li>
<li>&gt;95% is very good</li>
<li>&gt;80% is good</li>
<li>&gt;70% is ok</li>
<li>&lt;70% is poor to poorer.</li>
</ul></li>
<li>How contaminated is the bin/MAG?
<ul>
<li>Only one copy of each gene should be present (single-copy).</li>
<li>Are there any markers for other lineages present?</li>
</ul></li>
</ul>
<div id="checkm-conda" class="section level3" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> CheckM: Conda</h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Due to program version conflicts we will use the <code>checkm</code> conda environment for this section.</p>
<p>Open a new terminal and activate the <code>checkm</code> environment.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="11-Genome_binning.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> <span class="ex">usecheckm</span></span></code></pre></div>
<p>Ensure you are in the correct directory.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="11-Genome_binning.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/7-Binning/K1/</span></code></pre></div>
</div>
<div id="checkm-run" class="section level3" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> CheckM: run</h3>
<p><img src="figures/enter_key.png" width="10%" style="display: block; margin: auto;" /></p>
<p><code>CheckM</code> has many different commands. We will use one of the common workflows it provides called <code>lineage_wf</code>. This carries out five of its commands in a workflow (i.e. the next step uses output from the previous step).</p>
<ol style="list-style-type: decimal">
<li><code>tree</code> - Places bins in the reference genome tree. This reference tree comes with <code>CheckM</code>.</li>
<li><code>tree_qa</code> - Assess the phylogenetic markers found in each bin.</li>
<li><code>lineage_set</code> - Infers lineage-specific marker sets for each bin.</li>
<li><code>analyze</code> - Identifies marker genes in bins.</li>
<li><code>qa</code> - Assesses the bins for contamination and completeness.</li>
</ol>
<p>Run the <code>CheckM</code> command (this will take a while):</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="11-Genome_binning.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="ex">checkm</span> lineage_wf <span class="kw">\</span></span>
<span id="cb75-2"><a href="11-Genome_binning.html#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--tab_table</span> -f MAGS_checkm.tsv <span class="kw">\</span></span>
<span id="cb75-3"><a href="11-Genome_binning.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="ex">-x</span> fa <span class="kw">\</span></span>
<span id="cb75-4"><a href="11-Genome_binning.html#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="ex">bins/</span> checkm_output</span></code></pre></div>
<p>The options we used are:</p>
<ul>
<li><code>--tab_table</code> : Prints results to a tab separated table.</li>
<li><code>-f</code> : File name to print result to (if not specified results will go to stdout).</li>
<li><code>-x</code> : Suffix/extension of bin files. Other files are ignored in the specified bin directory.
<ul>
<li><code>fa</code> is used as our <code>MetaBAT2</code> produce fasta files end in <code>.fa</code>.</li>
</ul></li>
<li><code>bins</code> : The second last argument is the <code>bin_dir</code>, the directory containg all the bins in fasta format.</li>
<li><code>checkm_output</code> : The last argument is the directory to store the output to. This directory should not exist prior to running.</li>
</ul>
</div>
<div id="checkm-output" class="section level3" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> CheckM: output</h3>
<p><img src="figures/output.png" width="10%" style="display: block; margin: auto;" /></p>
<p>As we have only used a subset of data the results are not very good so let us look at premade results. These premade results were produced using the entire K1 dataset. First you will need to copy it over.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="11-Genome_binning.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/7-Binning</span>
<span id="cb76-2"><a href="11-Genome_binning.html#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> -s /pub39/tea/matther/NEOF/Shotgun_metagenomics/binning/K1_fullset/ .</span>
<span id="cb76-3"><a href="11-Genome_binning.html#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> K1_fullset</span></code></pre></div>
<p>Now we can look at the results table that is in your current directory.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="11-Genome_binning.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> MAGS_checkm.tsv</span></code></pre></div>
<p><strong>Tip</strong>: The definition of the statistics can be found at: <a href="https://github.com/Ecogenomics/CheckM/wiki/Reported-Statistics" class="uri">https://github.com/Ecogenomics/CheckM/wiki/Reported-Statistics</a></p>
<p>One quick way to calculate the overall quality of a bin is with the following equation:</p>
<p><span class="math display">\[
q = comp - (5 * cont)
\]</span>
Where:
<strong>q</strong> = Overall quality
<strong>comp</strong> = Completeness
<strong>cont</strong> = Contamination</p>
<p>A score of 70-80% would be the aim. Therefore let us calculate this for the bins with R.</p>
<p>First the tab delimited file has some issues for <code>R</code> so we’ll make a comma separated version.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="11-Genome_binning.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> MAGS_checkm.tsv <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&quot;\t&quot;</span> <span class="st">&quot;,&quot;</span> <span class="op">&gt;</span> MAGS_checkm.csv</span></code></pre></div>
<p>Activate <code>R</code>:</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="11-Genome_binning.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>Now to do the calculation in R for each bin:</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="11-Genome_binning.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in csv file as data frame</span></span>
<span id="cb80-2"><a href="11-Genome_binning.html#cb80-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;MAGS_checkm.csv&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb80-3"><a href="11-Genome_binning.html#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Add a column for overall quality</span></span>
<span id="cb80-4"><a href="11-Genome_binning.html#cb80-4" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>quality <span class="ot">&lt;-</span> df<span class="sc">$</span>Completeness <span class="sc">-</span> (<span class="dv">5</span> <span class="sc">*</span> df<span class="sc">$</span>Contamination)</span>
<span id="cb80-5"><a href="11-Genome_binning.html#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="co">#View the bin name, lineage, completeness, contamination and the quality columns</span></span>
<span id="cb80-6"><a href="11-Genome_binning.html#cb80-6" aria-hidden="true" tabindex="-1"></a>df[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">12</span>,<span class="dv">13</span>,<span class="dv">15</span>)]</span></code></pre></div>
<p><strong>Questions</strong>:</p>
<ul>
<li>What lineages were assigned to the bins?</li>
<li>How many genomes are present in the bins?</li>
<li>What is the completeness and contamination of the bins and how is the overall quality?</li>
<li>With this information how successful was the binning? Was the binning able to create MAGs (i.e. one genome per bin)?</li>
<li>Which bins look particularly good?</li>
</ul>
<p>Once you are happy you can quit <code>R</code>.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="11-Genome_binning.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">q</span>()</span></code></pre></div>
<p>It is always useful to know the quality of your bins so you know which are more reliable than others. With that information you can be more or less certain when concluding your findings.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="10-Metagenome_assembly.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="12-MAG_annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
