<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Functional annotation | Shotgun Metagenomics</title>
  <meta name="description" content="Chapter 12 Functional annotation | Shotgun Metagenomics" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Functional annotation | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Functional annotation | Shotgun Metagenomics" />
  
  
  

<meta name="author" content="Sam Haldenby and Matthew Gemmell" />


<meta name="date" content="2021-05-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="11-Gene_prediction.html"/>
<link rel="next" href="13-Co_assembly.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.1.1</b> Krona plot</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.2</b> Bracken</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.3</b> LEfSe biomarker detection</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2"><i class="fa fa-check"></i><b>9.1</b> HUMAnN2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#statistical-comparison-between-samples"><i class="fa fa-check"></i><b>9.2</b> Statistical comparison between samples</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#combining-data"><i class="fa fa-check"></i><b>9.2.1</b> Combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#renormalising-data"><i class="fa fa-check"></i><b>9.2.2</b> Renormalising data</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#visualisation"><i class="fa fa-check"></i><b>9.2.3</b> Visualisation</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#finding-statistically-significant-differences"><i class="fa fa-check"></i><b>9.2.4</b> Finding statistically significant differences</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.1</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.2</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.3</b> Assembly</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.3.1</b> QUAST</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Gene_prediction.html"><a href="11-Gene_prediction.html"><i class="fa fa-check"></i><b>11</b> Gene Prediciton</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Gene_prediction.html"><a href="11-Gene_prediction.html#running-the-software"><i class="fa fa-check"></i><b>11.1</b> Running the software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html"><i class="fa fa-check"></i><b>12</b> Functional annotation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#annotation-types"><i class="fa fa-check"></i><b>12.1</b> Annotation types</a></li>
<li class="chapter" data-level="12.2" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#running-cognizer"><i class="fa fa-check"></i><b>12.2</b> Running Cognizer</a></li>
<li class="chapter" data-level="12.3" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#visualisation-1"><i class="fa fa-check"></i><b>12.3</b> Visualisation</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#cog-categories"><i class="fa fa-check"></i><b>12.3.1</b> COG categories</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Co_assembly.html"><a href="13-Co_assembly.html"><i class="fa fa-check"></i><b>13</b> Co-assembly and coverage</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Co_assembly.html"><a href="13-Co_assembly.html#running-the-co-assembly-gene-predictions-and-annotations"><i class="fa fa-check"></i><b>13.1</b> Running the co-assembly, gene predictions and annotations</a></li>
<li class="chapter" data-level="13.2" data-path="13-Co_assembly.html"><a href="13-Co_assembly.html#generation-of-coverage-metrics"><i class="fa fa-check"></i><b>13.2</b> Generation of coverage metrics</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-Comparative_analysis.html"><a href="14-Comparative_analysis.html"><i class="fa fa-check"></i><b>14</b> Comparative analysis</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Comparative_analysis.html"><a href="14-Comparative_analysis.html#cog-categories-1"><i class="fa fa-check"></i><b>14.1</b> COG categories</a></li>
<li class="chapter" data-level="14.2" data-path="14-Comparative_analysis.html"><a href="14-Comparative_analysis.html#kegg-pathways-1"><i class="fa fa-check"></i><b>14.2</b> KEGG pathways</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Appendix.html"><a href="15-Appendix.html"><i class="fa fa-check"></i><b>15</b> Appendix</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Appendix.html"><a href="15-Appendix.html#cog-cateories"><i class="fa fa-check"></i><b>15.1</b> COG Cateories</a></li>
<li class="chapter" data-level="15.2" data-path="15-Appendix.html"><a href="15-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>15.2</b> Obtaining Read Data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-annotation" class="section level1" number="12">
<h1><span class="header-section-number">Chapter 12</span> Functional annotation</h1>
<p><img src="figures/quality_trimming_and_filtering.png" width="30%" style="display: block; margin: auto;" /></p>
<p>Now that we have a set of genes/proteins predicted in our metagenomic sample, we can functionally annotate them. As always, there are a variety of methods for obtaining functional annotations (e.g. rpsBLAST vs CDD/KOG, BLASTKoala).</p>
<p>Today we will use an all-in-one solution called Cognizer which assigns functions inferred from mapping to COG categorisations (KEGG, COG, SEED, GO, Pfam). This framework tool was actually intended for use with reads as input rather than gene/protein sequences: It will take reads and runs a translated BLAST (BLASTP) search against a custom protein database, before determining a variety of functional annotations for each read.</p>
<p>Unfortunately, this is a very processor intensive and slow process so we are cutting the workload down dramatically by instead providing the software with our predicted protein sequences. Even then, this process takes too long to complete in a tutorial session, so we will instead run the program on a subset of our data, i.e. 2000 proteins.</p>
<p>To do this, we will use Seqtk (Sequence tool kit)</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb69-1"><a href="12-Functional_annotation.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqtk</span> sample K1.newHeaders.faa 2000 <span class="op">&gt;</span> K1.sample.faa</span></code></pre></div>
<p>This command simply tells Seqtk to select 2000 entries from <code>K1.newHeaders.faa</code> and output them in <code>K1.sample.faa</code>. This resulting set of proteins will be much more manageable.</p>
<div id="annotation-types" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Annotation types</h2>
<p>Cognizer is a very useful tool because it assigns multiple different functional annotations in one run. Below is a brief description of these annotation types.</p>
<ul>
<li><strong>COG</strong>: Clusters of Orthologous Groups database generated by comparing predicted and known proteins in all completely sequenced microbial genomes.</li>
<li><strong>COG Functional Description</strong>: Based on single letter annotation, these descriptions categorise the function of the gene/protein. A table of these can be found in the appendix of this document.</li>
<li><strong>KEGG</strong>: A database resource for understanding high-level functions. Annotations range from small molecules up to whole biological pathways. Today we will touch upon KEGG Orthology (KO systems) which can be used to construct KEGG pathway modules (functional units) which are part of the larger KEGG pathways.</li>
<li><strong>Pfam</strong>: A database comprising a large collection of protein families, each represented by multiple sequence alignments and hidden Markov models.</li>
<li><strong>GO</strong>: Gene Ontology database that classifies functions along three aspects: molecular function (i.e. activities of gene products), cellular function (i.e. where the gene product is active) and biological process (e.g. pathways)</li>
<li><strong>FIG</strong>: Set of protein sequences that are similar along their full length (not just based on a domain). All of the proteins within a single FIG family (FIGfam) are believed to implement the same function</li>
<li><strong>SEED</strong>: Database created in 2004 for genome annotation annotation purposes, which predicts gene function and allows for the discovery of new pathways.</li>
</ul>
</div>
<div id="running-cognizer" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Running Cognizer</h2>
<p>Make a new directory to store the annotations in.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb70-1"><a href="12-Functional_annotation.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb70-2"><a href="12-Functional_annotation.html#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="12-Functional_annotation.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 8-GeneAnnotations</span>
<span id="cb70-4"><a href="12-Functional_annotation.html#cb70-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-5"><a href="12-Functional_annotation.html#cb70-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 8-GeneAnnotations</span>
<span id="cb70-6"><a href="12-Functional_annotation.html#cb70-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-7"><a href="12-Functional_annotation.html#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="ex">cognizer_aa</span> -d <span class="va">$DB</span>/Cognizer/ -e -10 <span class="kw">\</span></span>
<span id="cb70-8"><a href="12-Functional_annotation.html#cb70-8" aria-hidden="true" tabindex="-1"></a><span class="ex">-i</span> ../7-GenePredictions/K1.sample.faa -t 12 -o K1.sample</span></code></pre></div>
<p>Here, we have asked cognizer to annotate out proteins with 12 CPUs and output any BLAST matches with an e-value of 1x10-10 or lower to an output directory named K1.sample.</p>
<p>Once Cognizer has finished, see what files it has generated:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb71-1"><a href="12-Functional_annotation.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> K1.sample</span></code></pre></div>
<p>You will see there are six stat files which contain the number of incidences of proteins matching an annotation from each of the above databases. The one other file, assignments.txt, contains a table showing all of the query protein sequences and the annotations found for that protein.</p>
<p>Have a look in some of the .stat files (e.g. cog.stat) to see what the annotations look like. Take a look at assignments.txt too to see if you can match up the columns with the annotation types</p>
</div>
<div id="visualisation-1" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Visualisation</h2>
<p>Before we move on to the next part of the tutorial, we will visualise some of the functional annotations we have assigned to our metagenome.</p>
<div id="cog-categories" class="section level3" number="12.3.1">
<h3><span class="header-section-number">12.3.1</span> COG categories</h3>
<p>We can quickly create a graph to show the frequency of genes assigned to each of the COG categories. First, we need to tabulate and calculate the frequencies of the results which reside at the top of the K1.sample/cog.stat file.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="12-Functional_annotation.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="ex">getCogProportions.sh</span> K1.sample/cog.stat K1 <span class="op">&gt;</span> K1.sample.cog.tsv</span></code></pre></div>
<p>The second argument, K1, is to tell the script the ID of the sample for graphing purpose.</p>
<p>We can now draw our graph:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="12-Functional_annotation.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="ex">CogBarChart.R</span> -i K1.sample.cog.tsv -o K1.sample.cog.pdf</span>
<span id="cb73-2"><a href="12-Functional_annotation.html#cb73-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-3"><a href="12-Functional_annotation.html#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="ex">okular</span> K1.sample.cog.pdf <span class="kw">&amp;</span></span></code></pre></div>
<p>_<strong>PLOT HERE</strong></p>
<p>The bar-plot tells us the proportion of genes belonging to each COG functional category, which you can look up in the appendix of this document., but that’s not particularly informative unless we can compare it with something: We will come to this soon.</p>
<div id="kegg-pathways" class="section level4" number="12.3.1.1">
<h4><span class="header-section-number">12.3.1.1</span> KEGG pathways</h4>
<p>The above section told us the proportion of genes assigned to different COG functional categories, but nothing about the relationship between genes. To do this we will use the KEGG annotations that we obtained, but instead of just using a subset of the data, we will use annotations generated from the whole of sample K1, calculated prior to the workshop. Copy the full file over:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="12-Functional_annotation.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">$DB</span>/FullAnalysis2019/8-GeneAnnotations/K1.f/assignments.txt <span class="kw">\</span></span>
<span id="cb74-2"><a href="12-Functional_annotation.html#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="ex">K1.full.assignments.txt</span></span></code></pre></div>
<p>If you take a look at the file, you will see that the first column consists of the gene/protein id, and the fourth column consists of one or more KO (KEGG orthologue) identifiers. For now, these are the only column we need so we will trim the rest away.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="12-Functional_annotation.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="ex">getKeggFromAssignments.sh</span> <span class="kw">\</span></span>
<span id="cb75-2"><a href="12-Functional_annotation.html#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="ex">K1.full.assignments.txt</span> <span class="op">&gt;</span> K1.full.kegg.tsv</span></code></pre></div>
<p>This has created a two column file: gene id and KO id, which we can submit on the KEGG website. There, it will reconstruct functional pathways highlighting the presence or absence of the identified KEGG orthologues. Open Firefox</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="12-Functional_annotation.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> <span class="kw">&amp;</span></span></code></pre></div>
<p>and navigate to www.genome.jp/kegg/tool/map_pathway.html. Here, press the ‘Browse’ button and select your home directory from the right side panel (‘Places’). NB: Your home directory will start with nsc. Then, select 8-GeneAnnotations → K1.full.kegg.tsv, then press ‘Exec.’</p>
<p><strong>PLOT HERE</strong></p>
<p>After a few moments, a long list of pathways will appear which you can click on to view. The number in brackets after the pathway shows the number of KEGG orthologue matches identified in this pathway.</p>
<p>Try selecting some of the pathways and examine the pathway diagrams. Components marked in green have been identified in the sample and using this data we can identify complete and broken pathways of interest. See image below.</p>
<p><strong>PLOT HERE</strong></p>
<p>Ultimately, there is a lot of information here, but it only becomes fully useful if we can either make it more quantitative, compare it to other samples, or both. Currently, the data we have is not representative of what is in the sample beyond a qualitative measure, i.e. we know that Gene A might be present in K1, but we don’t know how much of Gene A is there. Once we have this information and the same information for K2, W1, and other samples we can start making some real comparisons. This will be covered in the following section.</p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="11-Gene_prediction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="13-Co_assembly.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
