<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Gene Prediciton | Shotgun Metagenomics</title>
  <meta name="description" content="Chapter 12 Gene Prediciton | Shotgun Metagenomics" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Gene Prediciton | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Gene Prediciton | Shotgun Metagenomics" />
  
  
  

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2021-05-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="11-Genome_binning.html"/>
<link rel="next" href="13-Functional_annotation.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-considerations"><i class="fa fa-check"></i><b>7.5</b> Host removal considerations</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-run"><i class="fa fa-check"></i><b>8.1.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-output"><i class="fa fa-check"></i><b>8.1.2</b> Kraken2: output</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.2</b> Krona plot</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.3</b> Bracken</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-run"><i class="fa fa-check"></i><b>8.3.1</b> Bracken: run</a></li>
<li class="chapter" data-level="8.3.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-output"><i class="fa fa-check"></i><b>8.3.2</b> Bracken: output</a></li>
<li class="chapter" data-level="8.3.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-merging-and-extracting-output"><i class="fa fa-check"></i><b>8.3.3</b> Bracken: merging and extracting output</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.4</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-add-metadata"><i class="fa fa-check"></i><b>8.4.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="8.4.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-conda"><i class="fa fa-check"></i><b>8.4.2</b> LEfSe: conda</a></li>
<li class="chapter" data-level="8.4.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-format"><i class="fa fa-check"></i><b>8.4.3</b> LEfSe: format</a></li>
<li class="chapter" data-level="8.4.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-run"><i class="fa fa-check"></i><b>8.4.4</b> LEfSe: run</a></li>
<li class="chapter" data-level="8.4.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-visualisation"><i class="fa fa-check"></i><b>8.4.5</b> LEfSe: visualisation</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>8.5</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2"><i class="fa fa-check"></i><b>9.1</b> HUMAnN2</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-conda-directories-and-files"><i class="fa fa-check"></i><b>9.1.1</b> HUMAnN2: conda, directories, and files</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-run"><i class="fa fa-check"></i><b>9.1.2</b> HUMAnN2: run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-output"><i class="fa fa-check"></i><b>9.1.3</b> HUMAnN2: output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#statistical-comparison-between-samples"><i class="fa fa-check"></i><b>9.2</b> Statistical comparison between samples</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#combining-data"><i class="fa fa-check"></i><b>9.2.1</b> Combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#renormalising-data"><i class="fa fa-check"></i><b>9.2.2</b> Renormalising data</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#visualisation"><i class="fa fa-check"></i><b>9.2.3</b> Visualisation</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#finding-statistically-significant-differences"><i class="fa fa-check"></i><b>9.2.4</b> Finding statistically significant differences</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.1</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.2</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.3</b> Assembly</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.3.1</b> QUAST</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html"><i class="fa fa-check"></i><b>11</b> Genome binning</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#genomic-binning-and-mags"><i class="fa fa-check"></i><b>11.1</b> Genomic binning and MAGs</a></li>
<li class="chapter" data-level="11.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#conda-environment-directory"><i class="fa fa-check"></i><b>11.2</b> Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#depth-calculation"><i class="fa fa-check"></i><b>11.3</b> Depth calculation</a></li>
<li class="chapter" data-level="11.4" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>11.4</b> MetaBAT2</a></li>
<li class="chapter" data-level="11.5" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>11.5</b> CheckM</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#conda"><i class="fa fa-check"></i><b>11.5.1</b> Conda</a></li>
<li class="chapter" data-level="11.5.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#running-checkm"><i class="fa fa-check"></i><b>11.5.2</b> Running CheckM</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Gene_prediction.html"><a href="12-Gene_prediction.html"><i class="fa fa-check"></i><b>12</b> Gene Prediciton</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Gene_prediction.html"><a href="12-Gene_prediction.html#running-the-software"><i class="fa fa-check"></i><b>12.1</b> Running the software</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Functional_annotation.html"><a href="13-Functional_annotation.html"><i class="fa fa-check"></i><b>13</b> Functional annotation</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Functional_annotation.html"><a href="13-Functional_annotation.html#annotation-types"><i class="fa fa-check"></i><b>13.1</b> Annotation types</a></li>
<li class="chapter" data-level="13.2" data-path="13-Functional_annotation.html"><a href="13-Functional_annotation.html#running-cognizer"><i class="fa fa-check"></i><b>13.2</b> Running Cognizer</a></li>
<li class="chapter" data-level="13.3" data-path="13-Functional_annotation.html"><a href="13-Functional_annotation.html#visualisation-1"><i class="fa fa-check"></i><b>13.3</b> Visualisation</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="13-Functional_annotation.html"><a href="13-Functional_annotation.html#cog-categories"><i class="fa fa-check"></i><b>13.3.1</b> COG categories</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-Co_assembly.html"><a href="14-Co_assembly.html"><i class="fa fa-check"></i><b>14</b> Co-assembly and coverage</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Co_assembly.html"><a href="14-Co_assembly.html#running-the-co-assembly-gene-predictions-and-annotations"><i class="fa fa-check"></i><b>14.1</b> Running the co-assembly, gene predictions and annotations</a></li>
<li class="chapter" data-level="14.2" data-path="14-Co_assembly.html"><a href="14-Co_assembly.html#generation-of-coverage-metrics"><i class="fa fa-check"></i><b>14.2</b> Generation of coverage metrics</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Comparative_analysis.html"><a href="15-Comparative_analysis.html"><i class="fa fa-check"></i><b>15</b> Comparative analysis</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Comparative_analysis.html"><a href="15-Comparative_analysis.html#cog-categories-1"><i class="fa fa-check"></i><b>15.1</b> COG categories</a></li>
<li class="chapter" data-level="15.2" data-path="15-Comparative_analysis.html"><a href="15-Comparative_analysis.html#kegg-pathways-1"><i class="fa fa-check"></i><b>15.2</b> KEGG pathways</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Appendix.html"><a href="16-Appendix.html"><i class="fa fa-check"></i><b>16</b> Appendix</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Appendix.html"><a href="16-Appendix.html#manuals"><i class="fa fa-check"></i><b>16.1</b> Manuals</a></li>
<li class="chapter" data-level="16.2" data-path="16-Appendix.html"><a href="16-Appendix.html#cog-cateories"><i class="fa fa-check"></i><b>16.2</b> COG Cateories</a></li>
<li class="chapter" data-level="16.3" data-path="16-Appendix.html"><a href="16-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>16.3</b> Obtaining Read Data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gene-prediciton" class="section level1" number="12">
<h1><span class="header-section-number">Chapter 12</span> Gene Prediciton</h1>
<p><img src="figures/quality_trimming_and_filtering.png" width="30%" style="display: block; margin: auto;" /></p>
<p>If we want to do functional analysis of our new metagenome assembly, we will first want to identify putative genes within it.</p>
<div id="running-the-software" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Running the software</h2>
<p>A popular tool for identifying genes (and therefore proteins) in assemblies based on sequence composition is Prodigal (Prokaryotic Dynamic Programming Genefinding Algorithm). Create a new directory and run Prodigal on your assembly:</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="12-Gene_prediction.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb77-2"><a href="12-Gene_prediction.html#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 7-GenePredictions</span>
<span id="cb77-3"><a href="12-Gene_prediction.html#cb77-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 7-GenePredictions</span>
<span id="cb77-4"><a href="12-Gene_prediction.html#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="ex">prodigal</span> -p meta <span class="kw">\</span></span>
<span id="cb77-5"><a href="12-Gene_prediction.html#cb77-5" aria-hidden="true" tabindex="-1"></a><span class="ex">-i</span> ../6-Assembly/K1/final.contigs.fa <span class="kw">\</span></span>
<span id="cb77-6"><a href="12-Gene_prediction.html#cb77-6" aria-hidden="true" tabindex="-1"></a><span class="ex">-a</span> K1.faa -d K1.fna -f gff -o K1.gff </span></code></pre></div>
<p>Here, we tell prodigal to run in metagenome mode (default is single genome mode). We also ask it to generate an amino acid FASTA file with protein predictions (K1.faa) in addition to our gene FASTA file (K1.fna). Finally, we instruct Prodigal to output a GFF (General Feature File), which contains all the gene prediction information.</p>
<p>After a few minutes, take a look at the output files</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="12-Gene_prediction.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>We have 3 new files, as requested.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="12-Gene_prediction.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> -S K1.gff </span>
<span id="cb79-2"><a href="12-Gene_prediction.html#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> K1.fna</span></code></pre></div>
<p>This file contains our predicted genes…</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="12-Gene_prediction.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> K1.faa</span></code></pre></div>
<p>…and this one contains our predicted proteins. For each gene, a corresponding protein with the same ID is predicted. In both of these files, you will notice that the headers are quite long and complex. This can interfere with downstream applications, so we’ll rename these now to avoid trouble later on</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="12-Gene_prediction.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="ex">RenameHeaders.py</span> --fasta K1.fna --name K1 <span class="dt">\ </span></span>
<span id="cb81-2"><a href="12-Gene_prediction.html#cb81-2" aria-hidden="true" tabindex="-1"></a>  <span class="ex">--zeros</span> 6 <span class="op">&gt;</span> K1.newHeaders.fna </span></code></pre></div>
<p>Now take a look at the new, re-headered gene FASTA file</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="12-Gene_prediction.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> K1.newHeaders.fna</span></code></pre></div>
<p>Much tidier.</p>
<p>Rename the headers in the protein FASTA file in a similar way to the command used above</p>
<p>Before we move on, we’ll check how many genes have been predicted in our mini-metagenome for Sample K1.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb83-1"><a href="12-Gene_prediction.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> -c <span class="st">&quot;^&gt;&quot;</span>  K1.newHeaders.fna</span></code></pre></div>
<p>This command searches for any line starting with &gt;. The <code>-c</code> flag tells <code>grep</code> to output the number of lines found with this pattern, i.e. the number of headers and therefore the number of genes.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="11-Genome_binning.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="13-Functional_annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
