<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Next steps | Shotgun Metagenomics</title>
  <meta name="description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="A Next steps | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Next steps | Shotgun Metagenomics" />
  
  <meta name="twitter:description" content="NEOF book for the Microbial shotgun metagenomics workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Sam Haldenby and Matthew R. Gemmell" />


<meta name="date" content="2021-06-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="13-MetaWrap.html"/>

<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Data &amp; QC</b></span></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
<li class="chapter" data-level="7.5" data-path="07-Host_removal.html"><a href="07-Host_removal.html#host-removal-considerations"><i class="fa fa-check"></i><b>7.5</b> Host removal considerations</a></li>
</ul></li>
<li class="part"><span><b>Read approach</b></span></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-run"><i class="fa fa-check"></i><b>8.1.1</b> Kraken2: run</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-output"><i class="fa fa-check"></i><b>8.1.2</b> Kraken2: output</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.2</b> Krona plot</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.3</b> Bracken</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-run"><i class="fa fa-check"></i><b>8.3.1</b> Bracken: run</a></li>
<li class="chapter" data-level="8.3.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-output"><i class="fa fa-check"></i><b>8.3.2</b> Bracken: output</a></li>
<li class="chapter" data-level="8.3.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-merging-output"><i class="fa fa-check"></i><b>8.3.3</b> Bracken: merging output</a></li>
<li class="chapter" data-level="8.3.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken-extracting-output"><i class="fa fa-check"></i><b>8.3.4</b> Bracken: extracting output</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.4</b> LEfSe biomarker detection</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-add-metadata"><i class="fa fa-check"></i><b>8.4.1</b> LEfSe: add metadata</a></li>
<li class="chapter" data-level="8.4.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-conda"><i class="fa fa-check"></i><b>8.4.2</b> LEfSe: conda</a></li>
<li class="chapter" data-level="8.4.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-format"><i class="fa fa-check"></i><b>8.4.3</b> LEfSe: format</a></li>
<li class="chapter" data-level="8.4.4" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-run"><i class="fa fa-check"></i><b>8.4.4</b> LEfSe: run</a></li>
<li class="chapter" data-level="8.4.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-visualisation"><i class="fa fa-check"></i><b>8.4.5</b> LEfSe: visualisation</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2-and-bracken-databases"><i class="fa fa-check"></i><b>8.5</b> Kraken2 and Bracken databases</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2"><i class="fa fa-check"></i><b>9.1</b> HUMAnN2</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-conda-directories-and-files"><i class="fa fa-check"></i><b>9.1.1</b> HUMAnN2: conda, directories, and files</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-run"><i class="fa fa-check"></i><b>9.1.2</b> HUMAnN2: run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-output"><i class="fa fa-check"></i><b>9.1.3</b> HUMAnN2: output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#statistical-comparison-between-samples"><i class="fa fa-check"></i><b>9.2</b> Statistical comparison between samples</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-combining-data"><i class="fa fa-check"></i><b>9.2.1</b> HUMAnN2 comparison: combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-renormalising-data"><i class="fa fa-check"></i><b>9.2.2</b> HUMAnN2 comparison: renormalising data</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-pca-plot"><i class="fa fa-check"></i><b>9.2.3</b> HUMAnN2 comparison: PCA plot</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2-comparison-lefse"><i class="fa fa-check"></i><b>9.2.4</b> HUMAnN2 comparison: LEfSe</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Assembly Approach</b></span></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#metagenome-assembly-conda"><i class="fa fa-check"></i><b>10.1</b> Metagenome assembly: Conda</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.2</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.3</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.4" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.4</b> Assembly</a></li>
<li class="chapter" data-level="10.5" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.5</b> QUAST</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html"><i class="fa fa-check"></i><b>11</b> Genome binning</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2"><i class="fa fa-check"></i><b>11.1</b> MetaBAT2</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-conda-environment-directory"><i class="fa fa-check"></i><b>11.1.1</b> MetaBAT2: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="11.1.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-depth-calculation"><i class="fa fa-check"></i><b>11.1.2</b> MetaBAT2: depth calculation</a></li>
<li class="chapter" data-level="11.1.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#metabat2-run"><i class="fa fa-check"></i><b>11.1.3</b> MetaBAT2: run</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm"><i class="fa fa-check"></i><b>11.2</b> CheckM</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-conda"><i class="fa fa-check"></i><b>11.2.1</b> CheckM: Conda</a></li>
<li class="chapter" data-level="11.2.2" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-run"><i class="fa fa-check"></i><b>11.2.2</b> CheckM: run</a></li>
<li class="chapter" data-level="11.2.3" data-path="11-Genome_binning.html"><a href="11-Genome_binning.html#checkm-output"><i class="fa fa-check"></i><b>11.2.3</b> CheckM: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html"><i class="fa fa-check"></i><b>12</b> Assembly functional annotation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#taxonomic-annotation"><i class="fa fa-check"></i><b>12.1</b> Taxonomic annotation</a></li>
<li class="chapter" data-level="12.2" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka"><i class="fa fa-check"></i><b>12.2</b> Prokka</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka-conda-and-directories"><i class="fa fa-check"></i><b>12.2.1</b> Prokka: conda and directories</a></li>
<li class="chapter" data-level="12.2.2" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka-run"><i class="fa fa-check"></i><b>12.2.2</b> Prokka: run</a></li>
<li class="chapter" data-level="12.2.3" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#prokka-output"><i class="fa fa-check"></i><b>12.2.3</b> Prokka: output</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath"><i class="fa fa-check"></i><b>12.3</b> MinPath</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-conda-and-directory"><i class="fa fa-check"></i><b>12.3.1</b> MinPath: conda and directory</a></li>
<li class="chapter" data-level="12.3.2" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-ec-extraction"><i class="fa fa-check"></i><b>12.3.2</b> MinPath: EC extraction</a></li>
<li class="chapter" data-level="12.3.3" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-run"><i class="fa fa-check"></i><b>12.3.3</b> MinPath: run</a></li>
<li class="chapter" data-level="12.3.4" data-path="12-Assembly_functional_annotation.html"><a href="12-Assembly_functional_annotation.html#minpath-output"><i class="fa fa-check"></i><b>12.3.4</b> MinPath: output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-MetaWrap.html"><a href="13-MetaWrap.html"><i class="fa fa-check"></i><b>13</b> MetaWRAP</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-MetaWrap.html"><a href="13-MetaWrap.html#tutorial"><i class="fa fa-check"></i><b>13.1</b> Tutorial</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="14-Appendix.html"><a href="14-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="14-Appendix.html"><a href="14-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="14-Appendix.html"><a href="14-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="next-steps" class="section level1" number="14">
<h1><span class="header-section-number">A</span> Next steps</h1>
<p><img src="figures/step.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Below are some good links to start with before carrying out your own projects.</p>
<ul>
<li>A review paper on metagenome assembly approaches
<ul>
<li>Genome-resolved metagenomics using environmental and clinical samples: <a href="https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbab030/6184411" class="uri">https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbab030/6184411</a></li>
</ul></li>
<li>bioBakery tools for meta’omic profiling
<ul>
<li><a href="https://github.com/biobakery/biobakery" class="uri">https://github.com/biobakery/biobakery</a></li>
</ul></li>
<li>Assessing the performance of different approaches for functional and taxonomic annotation of metagenomes
<ul>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6289-6" class="uri">https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6289-6</a></li>
</ul></li>
<li>MicrobeAnnotator: Easy-to-use pipeline for the comprehensive metabolic annotation of microbial genomes.
<ul>
<li><a href="https://github.com/cruizperez/MicrobeAnnotator" class="uri">https://github.com/cruizperez/MicrobeAnnotator</a></li>
</ul></li>
<li>MetaEuk: Functional annotation of eukaryotic metagenome
<ul>
<li><a href="https://github.com/soedinglab/metaeuk" class="uri">https://github.com/soedinglab/metaeuk</a>
_ Methods for Metagenomic data visualisation and analysis</li>
<li><a href="https://www.researchgate.net/publication/318252633_Methods_for_The_Metagenomic_Data_Visualization_and_Analysis" class="uri">https://www.researchgate.net/publication/318252633_Methods_for_The_Metagenomic_Data_Visualization_and_Analysis</a></li>
</ul></li>
</ul>
</div>
<div id="manuals" class="section level1" number="15">
<h1><span class="header-section-number">B</span> Manuals</h1>
<p><img src="figures/manual.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Conda: <a href="https://conda.io/projects/conda/en/latest/user-guide/getting-started.html" class="uri">https://conda.io/projects/conda/en/latest/user-guide/getting-started.html</a></p>
<p>FastQC: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
<p>MultiQC: <a href="https://multiqc.info/" class="uri">https://multiqc.info/</a></p>
<p>Trim Galore: <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/" class="uri">https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/</a></p>
<p>Bowtie2: <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml" class="uri">http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml</a></p>
<p>samtools: <a href="http://www.htslib.org/" class="uri">http://www.htslib.org/</a></p>
<p>BBTools: <a href="https://jgi.doe.gov/data-and-tools/bbtools/" class="uri">https://jgi.doe.gov/data-and-tools/bbtools/</a></p>
<p>Kraken2: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual</a></p>
<p>Krona: <a href="https://github.com/marbl/Krona/wiki/KronaTools" class="uri">https://github.com/marbl/Krona/wiki/KronaTools</a></p>
<p>Bracken: <a href="https://ccb.jhu.edu/software/bracken/index.shtml?t=manual" class="uri">https://ccb.jhu.edu/software/bracken/index.shtml?t=manual</a></p>
<p>LEfSe: <a href="https://huttenhower.sph.harvard.edu/lefse/" class="uri">https://huttenhower.sph.harvard.edu/lefse/</a></p>
<p>HUMAnN2: <a href="https://github.com/biobakery/biobakery/wiki/humann2" class="uri">https://github.com/biobakery/biobakery/wiki/humann2</a></p>
<p>MetaPhlAn2: <a href="https://huttenhower.sph.harvard.edu/metaphlan2/" class="uri">https://huttenhower.sph.harvard.edu/metaphlan2/</a></p>
<p>Biobakery: <a href="https://github.com/biobakery/biobakery" class="uri">https://github.com/biobakery/biobakery</a></p>
<p>MegaHit: <a href="https://github.com/voutcn/megahit" class="uri">https://github.com/voutcn/megahit</a></p>
<p>BWA: <a href="https://github.com/lh3/bwa" class="uri">https://github.com/lh3/bwa</a></p>
<p>minimap2: <a href="https://github.com/lh3/minimap2" class="uri">https://github.com/lh3/minimap2</a></p>
<p>MetaBAT2: <a href="https://bitbucket.org/berkeleylab/metabat/src/master/" class="uri">https://bitbucket.org/berkeleylab/metabat/src/master/</a></p>
<p>CheckM: <a href="https://github.com/Ecogenomics/CheckM/wiki" class="uri">https://github.com/Ecogenomics/CheckM/wiki</a></p>
<p>PhyloPhlAn: <a href="https://github.com/biobakery/phylophlan/wiki" class="uri">https://github.com/biobakery/phylophlan/wiki</a></p>
<p>MetaWRAP: <a href="https://github.com/bxlab/metaWRAP" class="uri">https://github.com/bxlab/metaWRAP</a></p>
<p>Pokka: <a href="https://github.com/tseemann/prokka" class="uri">https://github.com/tseemann/prokka</a></p>
<p>MinPath: <a href="https://github.com/mgtools/MinPath/blob/master/readme" class="uri">https://github.com/mgtools/MinPath/blob/master/readme</a></p>
<p>MetaCyc: <a href="https://metacyc.org/" class="uri">https://metacyc.org/</a></p>
</div>
<div id="obtaining-read-data" class="section level1" number="16">
<h1><span class="header-section-number">C</span> Obtaining Read Data</h1>
<p>The following commands can be used to obtain the sequence data used in this practical, directly from the EBI metagenomics site. It is worth noting that these are the full set of data, not like the miniaturised version you have used in the tutorial.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb100-1"><a href="14-Appendix.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_1.fastq.gz</span>
<span id="cb100-2"><a href="14-Appendix.html#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_2.fastq.gz</span>
<span id="cb100-3"><a href="14-Appendix.html#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K2_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_1.fastq.gz</span>
<span id="cb100-4"><a href="14-Appendix.html#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> K2_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_2.fastq.gz</span>
<span id="cb100-5"><a href="14-Appendix.html#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_1.fastq.gz</span>
<span id="cb100-6"><a href="14-Appendix.html#cb100-6" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> W1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_2.fastq.gz</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="13-MetaWrap.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
