<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 15 Appendix | Shotgun Metagenomics</title>
  <meta name="description" content="Chapter 15 Appendix | Shotgun Metagenomics" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 15 Appendix | Shotgun Metagenomics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 15 Appendix | Shotgun Metagenomics" />
  
  
  

<meta name="author" content="Sam Haldenby and Matthew Gemmell" />


<meta name="date" content="2021-05-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="14-Comparative_analysis.html"/>

<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Shotgun Metagenomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Shotgun_metagenomics.html"><a href="01-Shotgun_metagenomics.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Overview.html"><a href="02-Overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Overview.html"><a href="02-Overview.html#what-is-metagenomics"><i class="fa fa-check"></i><b>2.1</b> What is metagenomics?</a></li>
<li class="chapter" data-level="2.2" data-path="02-Overview.html"><a href="02-Overview.html#why-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Why metagenomics?</a></li>
<li class="chapter" data-level="2.3" data-path="02-Overview.html"><a href="02-Overview.html#metagenomics-vs-metagenetics"><i class="fa fa-check"></i><b>2.3</b> Metagenomics vs Metagenetics</a></li>
<li class="chapter" data-level="2.4" data-path="02-Overview.html"><a href="02-Overview.html#tutorial-overview"><i class="fa fa-check"></i><b>2.4</b> Tutorial overview</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="02-Overview.html"><a href="02-Overview.html#basics"><i class="fa fa-check"></i><b>2.4.1</b> Basics</a></li>
<li class="chapter" data-level="2.4.2" data-path="02-Overview.html"><a href="02-Overview.html#structure"><i class="fa fa-check"></i><b>2.4.2</b> Structure</a></li>
<li class="chapter" data-level="2.4.3" data-path="02-Overview.html"><a href="02-Overview.html#content"><i class="fa fa-check"></i><b>2.4.3</b> Content</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Before we start</a></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Raw_data.html"><a href="05-Raw_data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Raw_data.html"><a href="05-Raw_data.html#obtaining-the-data"><i class="fa fa-check"></i><b>5.1</b> Obtaining the data</a></li>
<li class="chapter" data-level="5.2" data-path="05-Raw_data.html"><a href="05-Raw_data.html#checking-quality-control"><i class="fa fa-check"></i><b>5.2</b> Checking quality control</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html"><i class="fa fa-check"></i><b>6</b> Quality control</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#removing-adapters-and-low-quality-bases"><i class="fa fa-check"></i><b>6.1</b> Removing adapters and low quality bases</a></li>
<li class="chapter" data-level="6.2" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#rename-the-files"><i class="fa fa-check"></i><b>6.2</b> Rename the files</a></li>
<li class="chapter" data-level="6.3" data-path="06-Trimming_data.html"><a href="06-Trimming_data.html#inspect-the-trimmed-data"><i class="fa fa-check"></i><b>6.3</b> Inspect the trimmed data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Host_removal.html"><a href="07-Host_removal.html"><i class="fa fa-check"></i><b>7</b> Host removal</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Host_removal.html"><a href="07-Host_removal.html#index-reference"><i class="fa fa-check"></i><b>7.1</b> Index reference</a></li>
<li class="chapter" data-level="7.2" data-path="07-Host_removal.html"><a href="07-Host_removal.html#alignment"><i class="fa fa-check"></i><b>7.2</b> Alignment</a></li>
<li class="chapter" data-level="7.3" data-path="07-Host_removal.html"><a href="07-Host_removal.html#unmapped-read-extraction"><i class="fa fa-check"></i><b>7.3</b> Unmapped read extraction</a></li>
<li class="chapter" data-level="7.4" data-path="07-Host_removal.html"><a href="07-Host_removal.html#re-pair"><i class="fa fa-check"></i><b>7.4</b> Re-pair</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html"><i class="fa fa-check"></i><b>8</b> Taxonomic profiling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#kraken2"><i class="fa fa-check"></i><b>8.1</b> Kraken2</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#krona-plot"><i class="fa fa-check"></i><b>8.1.1</b> Krona plot</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#bracken"><i class="fa fa-check"></i><b>8.2</b> Bracken</a></li>
<li class="chapter" data-level="8.3" data-path="08-Taxonomic_profiling.html"><a href="08-Taxonomic_profiling.html#lefse-biomarker-detection"><i class="fa fa-check"></i><b>8.3</b> LEfSe biomarker detection</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html"><i class="fa fa-check"></i><b>9</b> Functional profiling</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#humann2"><i class="fa fa-check"></i><b>9.1</b> HUMAnN2</a></li>
<li class="chapter" data-level="9.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#statistical-comparison-between-samples"><i class="fa fa-check"></i><b>9.2</b> Statistical comparison between samples</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#combining-data"><i class="fa fa-check"></i><b>9.2.1</b> Combining data</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#renormalising-data"><i class="fa fa-check"></i><b>9.2.2</b> Renormalising data</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#visualisation"><i class="fa fa-check"></i><b>9.2.3</b> Visualisation</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Functional_profiling.html"><a href="09-Functional_profiling.html#finding-statistically-significant-differences"><i class="fa fa-check"></i><b>9.2.4</b> Finding statistically significant differences</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html"><i class="fa fa-check"></i><b>10</b> Metagenome assembly</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#a-primer-on-short-read-assembly"><i class="fa fa-check"></i><b>10.1</b> A primer on short read assembly</a></li>
<li class="chapter" data-level="10.2" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#stitching-read-pairs"><i class="fa fa-check"></i><b>10.2</b> Stitching read pairs</a></li>
<li class="chapter" data-level="10.3" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#assembly"><i class="fa fa-check"></i><b>10.3</b> Assembly</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="10-Metagenome_assembly.html"><a href="10-Metagenome_assembly.html#quast"><i class="fa fa-check"></i><b>10.3.1</b> QUAST</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Gene_prediction.html"><a href="11-Gene_prediction.html"><i class="fa fa-check"></i><b>11</b> Gene Prediciton</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Gene_prediction.html"><a href="11-Gene_prediction.html#running-the-software"><i class="fa fa-check"></i><b>11.1</b> Running the software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html"><i class="fa fa-check"></i><b>12</b> Functional annotation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#annotation-types"><i class="fa fa-check"></i><b>12.1</b> Annotation types</a></li>
<li class="chapter" data-level="12.2" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#running-cognizer"><i class="fa fa-check"></i><b>12.2</b> Running Cognizer</a></li>
<li class="chapter" data-level="12.3" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#visualisation-1"><i class="fa fa-check"></i><b>12.3</b> Visualisation</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="12-Functional_annotation.html"><a href="12-Functional_annotation.html#cog-categories"><i class="fa fa-check"></i><b>12.3.1</b> COG categories</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Co_assembly.html"><a href="13-Co_assembly.html"><i class="fa fa-check"></i><b>13</b> Co-assembly and coverage</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Co_assembly.html"><a href="13-Co_assembly.html#running-the-co-assembly-gene-predictions-and-annotations"><i class="fa fa-check"></i><b>13.1</b> Running the co-assembly, gene predictions and annotations</a></li>
<li class="chapter" data-level="13.2" data-path="13-Co_assembly.html"><a href="13-Co_assembly.html#generation-of-coverage-metrics"><i class="fa fa-check"></i><b>13.2</b> Generation of coverage metrics</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="14-Comparative_analysis.html"><a href="14-Comparative_analysis.html"><i class="fa fa-check"></i><b>14</b> Comparative analysis</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Comparative_analysis.html"><a href="14-Comparative_analysis.html#cog-categories-1"><i class="fa fa-check"></i><b>14.1</b> COG categories</a></li>
<li class="chapter" data-level="14.2" data-path="14-Comparative_analysis.html"><a href="14-Comparative_analysis.html#kegg-pathways-1"><i class="fa fa-check"></i><b>14.2</b> KEGG pathways</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Appendix.html"><a href="15-Appendix.html"><i class="fa fa-check"></i><b>15</b> Appendix</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Appendix.html"><a href="15-Appendix.html#cog-cateories"><i class="fa fa-check"></i><b>15.1</b> COG Cateories</a></li>
<li class="chapter" data-level="15.2" data-path="15-Appendix.html"><a href="15-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>15.2</b> Obtaining Read Data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Shotgun Metagenomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="appendix" class="section level1" number="15">
<h1><span class="header-section-number">Chapter 15</span> Appendix</h1>
<p><img src="figures/quality_trimming_and_filtering.png" width="30%" style="display: block; margin: auto;" /></p>
<p>##Manuals</p>
<p>Kraken2: <a href="https://github.com/DerrickWood/kraken2/wiki/Manual" class="uri">https://github.com/DerrickWood/kraken2/wiki/Manual</a>
Krona: <a href="https://github.com/marbl/Krona/wiki/KronaTools" class="uri">https://github.com/marbl/Krona/wiki/KronaTools</a>
Bracken: <a href="https://ccb.jhu.edu/software/bracken/index.shtml?t=manual" class="uri">https://ccb.jhu.edu/software/bracken/index.shtml?t=manual</a>
LEfSe: <a href="https://huttenhower.sph.harvard.edu/lefse/" class="uri">https://huttenhower.sph.harvard.edu/lefse/</a>
HUMAnN2: <a href="https://github.com/biobakery/biobakery/wiki/humann2" class="uri">https://github.com/biobakery/biobakery/wiki/humann2</a>
Biobakery: <a href="https://github.com/biobakery/biobakery" class="uri">https://github.com/biobakery/biobakery</a></p>
<div id="cog-cateories" class="section level2" number="15.1">
<h2><span class="header-section-number">15.1</span> COG Cateories</h2>
<ul>
<li>CELLULAR PROCESSES AND SIGNALING
<ul>
<li>[D] Cell cycle control, cell division, chromosome partitioning</li>
<li>[M] Cell wall/membrane/envelope biogenesis</li>
<li>[N] Cell motility</li>
<li>[O] Post-translational modification, protein turnover, and chaperones</li>
<li>[T] Signal transduction mechanisms</li>
<li>[U] Intracellular trafficking, secretion, and vesicular transport</li>
<li>[V] Defense mechanisms</li>
<li>[W] Extracellular structures</li>
<li>[Y] Nuclear structure</li>
<li>[Z] Cytoskeleton</li>
</ul></li>
<li>INFORMATION STORAGE AND PROCESSING
<ul>
<li>[A] RNA processing and modification</li>
<li>[B] Chromatin structure and dynamics</li>
<li>[J] Translation, ribosomal structure and biogenesis</li>
<li>[K] Transcription</li>
<li>[L] Replication, recombination and repair</li>
</ul></li>
<li>METABOLISM
<ul>
<li>[C] Energy production and conversion</li>
<li>[E] Amino acid transport and metabolism</li>
<li>[F] Nucleotide transport and metabolism</li>
<li>[G] Carbohydrate transport and metabolism</li>
<li>[H] Coenzyme transport and metabolism</li>
<li>[I] Lipid transport and metabolism</li>
<li>[P] Inorganic ion transport and metabolism</li>
<li>[Q] Secondary metabolites biosynthesis, transport, and catabolism</li>
</ul></li>
<li>POORLY CHARACTERIZED
<ul>
<li>[R] General function prediction only</li>
<li>[S] Function unknown</li>
</ul></li>
</ul>
</div>
<div id="obtaining-read-data" class="section level2" number="15.2">
<h2><span class="header-section-number">15.2</span> Obtaining Read Data</h2>
<p>The following commands can be used to obtain the sequence data used in this practical, directly from the EBI metagenomics site. It is worth noting that these are the full set of data, not like the miniaturised version you have used in the tutorial.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb99-1"><a href="15-Appendix.html#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -O K1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_1.fastq.gz</span>
<span id="cb99-2"><a href="15-Appendix.html#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -O K1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505102/ERR505102_2.fastq.gz</span>
<span id="cb99-3"><a href="15-Appendix.html#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -O K2_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_1.fastq.gz</span>
<span id="cb99-4"><a href="15-Appendix.html#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -O K2_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505104/ERR505104_2.fastq.gz</span>
<span id="cb99-5"><a href="15-Appendix.html#cb99-5" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -O W1_R1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_1.fastq.gz</span>
<span id="cb99-6"><a href="15-Appendix.html#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -O W1_R2.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR505/ERR505090/ERR505090_2.fastq.gz</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="14-Comparative_analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
